// vite-plugin-rnw@0.0.11/dist/index.mjs
import * as vite from "vite";
import { createFilter, version } from "vite";
import commonjs from "vite-plugin-commonjs";
import { makeIdFiltersToMatchWithQuery } from "@rolldown/pluginutils";
import MagicString from "magic-string";
import react from "@vitejs/plugin-react";
import { readFile } from "node:fs/promises";
import flowRemoveTypes from "flow-remove-types";
//#region src/transforms.ts
/**
* Code transformation utilities for fixing React Native package compatibility issues
*/
/**
* Analyzes require statements in code and maps them to module imports.
* Handles patterns like: varName = require('path').prop or require('path').default
*/
function analyzeRequireStatements(originalCode, exportedVars) {
	const importMap = /* @__PURE__ */ new Map();
	for (const varName of exportedVars) {
		const requireMatch = new RegExp(`${varName}\\s*=\\s*[\\s\\S]*?require\\(['"]([^'"]+)['"]\\)(?:\\.([\\w]+))?`, "g").exec(originalCode);
		if (requireMatch) {
			const [, modulePath, prop] = requireMatch;
			const key = `${modulePath}:${prop || "default"}`;
			if (!importMap.has(key)) importMap.set(key, []);
			importMap.get(key).push(varName);
		}
	}
	return importMap;
}
/**
* Removes try/catch blocks that contain require statements and export let declarations.
* This cleans up the problematic CommonJS patterns.
*/
/**
* Generates direct export statements from the import map.
* Creates modern ESM exports like: export { default as varName } from 'module';
*/
function generateDirectExports(importMap) {
	let exports = "";
	for (const [key, vars] of importMap) {
		const [modulePath, prop] = key.split(":");
		if (prop === "default") for (const varName of vars) exports += `export { default as ${varName} } from '${modulePath}';\n`;
		else for (const varName of vars) exports += `export { ${prop} as ${varName} } from '${modulePath}';\n`;
	}
	return exports;
}
/**
* Transforms React Native Reanimated webUtils files to fix problematic
* export let + try/catch + require patterns by converting them to proper ESM.
*
* This transformation handles the incompatible module patterns in React Native Reanimated
* that cause build errors in Vite due to mixing CommonJS require() with ESM exports.
*/
function transformReanimatedWebUtilsWithMap(code, id, isProduction, opts) {
	if (!isProduction || !id.includes("node_modules/react-native-reanimated") || !id.includes("ReanimatedModule/js-reanimated/webUtils") || !code.includes("export let") || !code.includes("try") || !code.includes("require")) return {
		code,
		map: null,
		changed: false
	};
	const exportedVars = Array.from(code.matchAll(/export let (\w+);/g)).map((match) => match[1]);
	if (exportedVars.length === 0) return {
		code,
		map: null,
		changed: false
	};
	const importMap = analyzeRequireStatements(code, exportedVars);
	const ms = new MagicString(code);
	let changed = false;
	for (const m of code.matchAll(/try\s*\{[^{}]*?require\([^)]+\)[^{}]*?\}\s*catch[^{}]*?\{[^{}]*?\}/gs)) {
		const start = m.index;
		const end = start + m[0].length;
		ms.remove(start, end);
		changed = true;
	}
	for (const m of code.matchAll(/export let \w+;/g)) {
		const start = m.index;
		const end = start + m[0].length;
		ms.remove(start, end);
		changed = true;
	}
	const exports = generateDirectExports(importMap);
	if (exports) {
		ms.append(`\n${exports}`);
		changed = true;
	}
	if (!changed) return {
		code,
		map: null,
		changed: false
	};
	return {
		code: ms.toString(),
		map: ms.generateMap({
			source: (opts === null || opts === void 0 ? void 0 : opts.source) ?? id.split("?")[0],
			includeContent: true,
			hires: true
		}),
		changed: true
	};
}
/**
* Transforms react-native-css-interop doctor check to always return true.
* Looks for:
*   return <react-native-css-interop-jsx-pragma-check /> === true;
* in node_modules/react-native-css-interop/dist/doctor.js and replaces with:
*   return true;
*/
function transformCssInteropDoctorCheck(code, id, opts) {
	if (!id.includes("node_modules/react-native-css-interop/dist/doctor.js")) return {
		code,
		map: null,
		changed: false
	};
	const pattern = /return\s*<react-native-css-interop-jsx-pragma-check\s*\/>\s*===\s*true\s*;/g;
	if (!pattern.test(code)) return {
		code,
		map: null,
		changed: false
	};
	const ms = new MagicString(code);
	let match;
	pattern.lastIndex = 0;
	while ((match = pattern.exec(code)) !== null) {
		const start = match.index;
		const end = start + match[0].length;
		ms.overwrite(start, end, "return true;");
	}
	return {
		code: ms.toString(),
		map: ms.generateMap({
			source: (opts === null || opts === void 0 ? void 0 : opts.source) ?? id.split("?")[0],
			includeContent: true,
			hires: true
		}),
		changed: true
	};
}
//#endregion
//#region src/removeFlow.ts
const defaultFlowOptions = {
	all: false,
	pretty: false,
	ignoreUninitializedFields: false
};
const defaultPluginOptions = {
	include: /\.(flow|jsx?)$/,
	exclude: /node_modules/,
	flow: defaultFlowOptions
};
const jsxRegex = /\.jsx$/;
const defaultLoaderFunction = (path) => jsxRegex.test(path) ? "jsx" : "js";
const toError = (error) => error instanceof Error ? error : new Error(String(error));
/**
* Create a Vite plugin object.
*/
function flowPlugin({ exclude = defaultPluginOptions.exclude, include = defaultPluginOptions.include, flow = defaultFlowOptions } = defaultPluginOptions) {
	const filter = createFilter(include, exclude);
	return {
		enforce: "pre",
		name: "flow",
		transform(src, id) {
			if (!filter(id)) return null;
			return {
				code: flowRemoveTypes(src, flow).toString(),
				map: null
			};
		}
	};
}
/**
* Create an esbuild plugin object.
*/
function esbuildFlowPlugin(filter = /\.(flow|jsx?)$/, loaderFunction = defaultLoaderFunction, flowOptions = defaultFlowOptions) {
	return {
		name: "flow",
		setup(build) {
			build.onLoad({ filter }, async ({ path, namespace }) => {
				try {
					return {
						contents: flowRemoveTypes(await readFile(path, "utf-8"), flowOptions).toString(),
						loader: loaderFunction(path)
					};
				} catch (error) {
					const resolvedError = toError(error);
					return { errors: [{
						text: resolvedError.message,
						location: {
							file: path,
							namespace
						},
						detail: resolvedError
					}] };
				}
			});
		}
	};
}
function rollDownFlowPlugin(filter = /\.(flow|jsx?)$/, loaderFunction = defaultLoaderFunction, flowOptions = defaultFlowOptions) {
	return {
		name: "rolldown-flow",
		load: {
			filter: { id: { include: filter } },
			handler: async (id) => {
				return {
					code: flowRemoveTypes(await readFile(id, "utf-8"), flowOptions).toString(),
					moduleType: loaderFunction(id)
				};
			}
		}
	};
}
//#endregion
//#region src/index.ts
const shouldUseRolldownOptions = () => {
	try {
		return Number(version.split(".")[0]) >= 8;
	} catch {
		return false;
	}
};
const shouldUseRollDown = shouldUseRolldownOptions();
const extensions = [
	".web.mjs",
	".web.js",
	".web.mts",
	".web.ts",
	".web.tsx",
	".web.cjs",
	".mjs",
	".js",
	".jsx",
	".json",
	".mts",
	".ts",
	".tsx",
	".cjs"
];
const defaultIncludeRE = /\.[tj]sx?$/;
const defaultExcludeRE = /\/node_modules\/(?!react-native|@react-native|expo|@expo)/;
const getJsxOption = (jsxRuntime) => {
	const jsxOptionMapping = {
		automatic: "automatic",
		classic: "transform"
	};
	return jsxRuntime && jsxRuntime in jsxOptionMapping ? jsxOptionMapping[jsxRuntime] : "automatic";
};
const treeshakePlugin = {
	name: "treeshake-fix",
	async transform(_code, id) {
		if (id.includes("react-native-css-interop") || id.includes("react-native-css") || id.includes("expo-modules-core")) return { moduleSideEffects: "no-treeshake" };
	}
};
const treeshakeSafePreset = {
	annotations: true,
	invalidImportSideEffects: true,
	manualPureFunctions: [],
	moduleSideEffects: true,
	propertyReadSideEffects: "always",
	unknownGlobalSideEffects: true,
	propertyWriteSideEffects: "always"
};
function getBuildOptions() {
	if (shouldUseRollDown) return { rolldownOptions: {
		shimMissingExports: true,
		treeshake: treeshakeSafePreset,
		plugins: [treeshakePlugin]
	} };
	return { rollupOptions: {
		shimMissingExports: true,
		treeshake: treeshakeSafePreset,
		plugins: [treeshakePlugin]
	} };
}
function getOptimizeDepsOptions(opts) {
	if (shouldUseRollDown) return { rolldownOptions: {
		resolve: { extensions },
		transform: { jsx: {
			importSource: opts.jsxImportSource,
			runtime: opts.jsxRuntime
		} },
		moduleTypes: {
			".js": "jsx",
			".mjs": "jsx"
		},
		plugins: [rollDownFlowPlugin(/* @__PURE__ */ new RegExp(/\.(flow|jsx?)$/), (_path) => "jsx")]
	} };
	return { esbuildOptions: {
		resolveExtensions: extensions,
		jsx: getJsxOption(opts.jsxRuntime),
		jsxImportSource: opts.jsxImportSource,
		loader: {
			".js": "jsx",
			".mjs": "jsx"
		},
		plugins: [esbuildFlowPlugin(/* @__PURE__ */ new RegExp(/\.(flow|jsx?)$/), (_path) => "jsx")]
	} };
}
function rnw(opts = {}) {
	const include = opts.include ?? defaultIncludeRE;
	const exclude = opts.exclude ?? defaultExcludeRE;
	const filter = createFilter(include, exclude);
	let isProduction = true;
	const rnwPlugin = {
		name: "vite:react-native-web-babel",
		enforce: "pre",
		config(_userConfig, { mode }) {
			const development = mode === "development";
			return {
				define: {
					global: "window",
					DEV: JSON.stringify(development),
					"global.__x": {},
					_frameTimestamp: void 0,
					_WORKLET: false,
					__DEV__: JSON.stringify(development),
					"process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV || mode),
					EXPO_OS: JSON.stringify("web"),
					"process.env.EXPO_OS": JSON.stringify("web"),
					"global.Error": "Error"
				},
				optimizeDeps: getOptimizeDepsOptions(opts),
				build: getBuildOptions(),
				resolve: {
					extensions,
					alias: { "react-native": "react-native-web" }
				}
			};
		},
		configResolved(config) {
			isProduction = config.isProduction;
		},
		transform: {
			filter: { id: {
				include: makeIdFiltersToMatchWithQuery(include),
				exclude: makeIdFiltersToMatchWithQuery(exclude)
			} },
			async handler(code, id, options) {
				const [filepath] = id.split("?");
				if (!filter(filepath)) return;
				const cssInteropResult = transformCssInteropDoctorCheck(code, id, { source: filepath });
				if (cssInteropResult.changed) return {
					code: cssInteropResult.code,
					map: cssInteropResult.map
				};
				const transformResult = transformReanimatedWebUtilsWithMap(code, id, isProduction, { source: filepath });
				if (!transformResult.changed) return null;
				return {
					code: transformResult.code,
					map: transformResult.map
				};
			}
		}
	};
	return [
		flowPlugin({ exclude }),
		{
			name: "treat-js-files-as-jsx",
			enforce: "pre",
			async transform(code, id) {
				if (id.match(/\.js$/) || id.match(/\.mjs$/)) {
					const jsxOption = getJsxOption(opts.jsxRuntime);
					if (shouldUseRollDown) return vite.transformWithOxc(code, id, {
						lang: "jsx",
						jsx: {
							importSource: opts.jsxImportSource,
							runtime: opts.jsxRuntime
						}
					});
					return vite.transformWithEsbuild(code, id, {
						loader: "jsx",
						jsx: jsxOption,
						jsxImportSource: opts.jsxImportSource
					});
				}
				return null;
			}
		},
		commonjs(),
		rnwPlugin,
		react({
			...opts,
			exclude
		})
	];
}
//#endregion
export { rnw };
