// @storybook/react-native-ui-common@10.4.7/dist/index.js
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
//#endregion
let _storybook_react_native_theming = require("@storybook/react-native-theming");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
let react_native = require("react-native");
let storybook_internal_csf = require("storybook/internal/csf");
let ts_dedent = require("ts-dedent");
let storybook_internal_client_logger = require("storybook/internal/client-logger");
let es_toolkit_array = require("es-toolkit/array");
let es_toolkit_object = require("es-toolkit/object");
let es_toolkit_predicate = require("es-toolkit/predicate");
let memoizerific = require("memoizerific");
memoizerific = __toESM(memoizerific);
//#region src/Button.tsx
const Button = (0, react.forwardRef)(({ Icon, animation = "none", size = "small", variant = "outline", padding = "medium", disabled = false, active = false, onPress, children, text, ...props }, ref) => {
	const [isAnimating, setIsAnimating] = (0, react.useState)(false);
	const handleClick = (event) => {
		if (onPress) onPress(event);
		if (animation === "none") return;
		setIsAnimating(true);
	};
	(0, react.useEffect)(() => {
		const timer = setTimeout(() => {
			if (isAnimating) setIsAnimating(false);
		}, 1e3);
		return () => clearTimeout(timer);
	}, [isAnimating]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(StyledButton, {
		ref,
		variant,
		size,
		padding,
		disabled,
		active,
		animating: isAnimating,
		animation,
		onPress: handleClick,
		accessibilityRole: "button",
		...props,
		children: [
			Icon && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(ButtonIcon, {
				Icon,
				variant,
				active
			}),
			text && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(ButtonText, {
				variant,
				active,
				children: text
			}),
			children
		]
	});
});
Button.displayName = "Button";
const StyledButton = _storybook_react_native_theming.styled.TouchableOpacity(({ theme, variant, size, disabled, active, padding }) => ({
	border: 0,
	display: "flex",
	flexDirection: "row",
	gap: 6,
	alignItems: "center",
	justifyContent: "center",
	overflow: "hidden",
	paddingHorizontal: (() => {
		if (padding === "small" && size === "small") return 7;
		if (padding === "small" && size === "medium") return 9;
		if (size === "small") return 10;
		if (size === "medium") return 12;
		return 0;
	})(),
	paddingVertical: 0,
	height: size === "small" ? 28 : 32,
	position: "relative",
	transitionProperty: "background, box-shadow",
	transitionDuration: "150ms",
	transitionTimingFunction: "ease-out",
	whiteSpace: "nowrap",
	userSelect: "none",
	opacity: disabled ? .5 : 1,
	margin: 0,
	backgroundColor: (() => {
		if (variant === "solid") return theme.color.secondary;
		if (variant === "outline") return theme.button.background;
		if (variant === "ghost" && active) return theme.background.hoverable;
		return "transparent";
	})(),
	boxShadow: variant === "outline" ? `${theme.button.border} 0 0 0 1px inset` : "none",
	borderRadius: theme.input.borderRadius,
	flexShrink: 0
}));
const ButtonText = _storybook_react_native_theming.styled.Text(({ theme, variant, active }) => ({
	color: (() => {
		if (variant === "solid") return theme.color.lightest;
		if (variant === "outline") return theme.input.color;
		if (variant === "ghost" && active) return theme.color.secondary;
		if (variant === "ghost") return theme.color.mediumdark;
		return theme.input.color;
	})(),
	flexDirection: "row",
	gap: 6,
	textAlign: "center",
	fontSize: theme.typography.size.s1,
	fontWeight: theme.typography.weight.bold
}));
const ButtonIcon = ({ Icon, active, variant }) => {
	const theme = (0, _storybook_react_native_theming.useTheme)();
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(Icon, { color: (0, react.useMemo)(() => {
		if (variant === "solid") return theme.color.lightest;
		if (variant === "outline") return theme.input.color;
		if (variant === "ghost" && active) return theme.color.secondary;
		if (variant === "ghost") return theme.color.mediumdark;
		return theme.input.color;
	}, [
		active,
		theme.color.lightest,
		theme.color.mediumdark,
		theme.color.secondary,
		theme.input.color,
		variant
	]) });
};
//#endregion
//#region src/IconButton.tsx
const IconButton = (0, react.forwardRef)(({ padding = "small", variant = "ghost", ...props }, ref) => {
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(Button, {
		padding,
		variant,
		ref,
		...props
	});
});
IconButton.displayName = "IconButton";
//#endregion
//#region src/LayoutProvider.tsx
const LayoutContext = (0, react.createContext)({
	isDesktop: false,
	isMobile: true
});
const LayoutProvider = ({ children }) => {
	const { width } = (0, react_native.useWindowDimensions)();
	const isDesktop = react_native.Platform.OS === "macos" || react_native.Platform.OS === "windows" || width >= 1e3;
	const isMobile = !isDesktop;
	const contextValue = (0, react.useMemo)(() => ({
		isDesktop,
		isMobile
	}), [isDesktop, isMobile]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(LayoutContext.Provider, {
		value: contextValue,
		children
	});
};
const useLayout = () => (0, react.useContext)(LayoutContext);
//#endregion
//#region src/types.ts
function isExpandType(x) {
	return !!(x && x.showAll);
}
//#endregion
//#region src/StorageProvider.tsx
const StorageContext = (0, react.createContext)({
	getItem: async () => null,
	setItem: async () => {}
});
const StorageProvider = ({ storage, children }) => {
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(StorageContext.Provider, {
		value: storage,
		children
	});
};
const useStorage = () => (0, react.useContext)(StorageContext);
//#endregion
//#region src/util/StoryHash.ts
const intersect = (a, b) => {
	if (!Array.isArray(a) || !Array.isArray(b) || !a.length || !b.length) return [];
	return a.reduce((acc, aValue) => {
		if (b.includes(aValue)) acc.push(aValue);
		return acc;
	}, []);
};
const merge = (a, ...b) => {
	let target = {};
	target = (0, es_toolkit_object.mergeWith)({}, a, (objValue, srcValue) => {
		if (Array.isArray(srcValue) && Array.isArray(objValue)) {
			srcValue.forEach((s) => {
				if (!objValue.find((o) => o === s || (0, es_toolkit_predicate.isEqual)(o, s))) objValue.push(s);
			});
			return objValue;
		}
		if (Array.isArray(objValue)) {
			storybook_internal_client_logger.logger.log(["the types mismatch, picking", objValue]);
			return objValue;
		}
	});
	for (const obj of b) target = (0, es_toolkit_object.mergeWith)(target, obj, (objValue, srcValue) => {
		if (Array.isArray(srcValue) && Array.isArray(objValue)) {
			srcValue.forEach((s) => {
				if (!objValue.find((o) => o === s || (0, es_toolkit_predicate.isEqual)(o, s))) objValue.push(s);
			});
			return objValue;
		}
		if (Array.isArray(objValue)) {
			storybook_internal_client_logger.logger.log(["the types mismatch, picking", objValue]);
			return objValue;
		}
	});
	return target;
};
const noArrayMerge = (a, ...b) => {
	let target = {};
	target = (0, es_toolkit_object.mergeWith)({}, a, (objValue, srcValue) => {
		if (Array.isArray(srcValue)) return srcValue;
	});
	for (const obj of b) target = (0, es_toolkit_object.mergeWith)(target, obj, (objValue, srcValue) => {
		if (Array.isArray(srcValue)) return srcValue;
	});
	return target;
};
const TITLE_PATH_SEPARATOR = /\s*\/\s*/;
const transformStoryIndexToStoriesHash = (input, { provider, docsOptions, filters, allStatuses }) => {
	if (!input.v) throw new Error("Composition: Missing stories.json version");
	let index = input;
	index = index.v === 2 ? transformStoryIndexV2toV3(index) : index;
	index = index.v === 3 ? transformStoryIndexV3toV4(index) : index;
	index = index.v === 4 ? transformStoryIndexV4toV5(index) : index;
	index = index;
	const entryValues = Object.values(index.entries).filter((entry) => {
		let result = true;
		const storyStatuses = allStatuses[entry.id] ?? {};
		if (Object.values(storyStatuses).some(({ value }) => value === "status-value:error")) return result;
		Object.values(filters).forEach((filter) => {
			if (result === false) return;
			result = filter({
				...entry,
				statuses: storyStatuses
			});
		});
		return result;
	});
	const { sidebar = {} } = provider.getConfig();
	const { showRoots, collapsedRoots = [], renderLabel } = sidebar;
	const setShowRoots = typeof showRoots !== "undefined";
	const storiesHashOutOfOrder = entryValues.reduce((acc, item) => {
		if (docsOptions.docsMode && item.type !== "docs") return acc;
		const { title } = item;
		const groups = title.trim().split(TITLE_PATH_SEPARATOR);
		const root = (!setShowRoots || showRoots) && groups.length > 1 ? [groups.shift()] : [];
		const names = [...root, ...groups];
		const paths = names.reduce((list, name, idx) => {
			const parent = idx > 0 && list[idx - 1];
			const id = (0, storybook_internal_csf.sanitize)(parent ? `${parent}-${name}` : name);
			if (name.trim() === "") throw new Error(ts_dedent.dedent`Invalid title ${title} ending in slash.`);
			if (parent === id) throw new Error(ts_dedent.dedent`
          Invalid part '${name}', leading to id === parentId ('${id}'), inside title '${title}'

          Did you create a path that uses the separator char accidentally, such as 'Vue <docs/>' where '/' is a separator char? See https://github.com/storybookjs/storybook/issues/6128
          `);
			list.push(id);
			return list;
		}, []);
		paths.forEach((id, idx) => {
			const childId = paths[idx + 1] || item.id;
			if (root.length && idx === 0) acc[id] = merge(acc[id] || {}, {
				type: "root",
				id,
				name: names[idx],
				tags: [],
				depth: idx,
				renderLabel,
				startCollapsed: collapsedRoots.includes(id),
				children: [childId]
			});
			else if ((!acc[id] || acc[id].type === "component") && idx === paths.length - 1) acc[id] = merge(acc[id] || {}, {
				type: "component",
				id,
				name: names[idx],
				tags: [],
				parent: paths[idx - 1],
				depth: idx,
				renderLabel,
				...childId && { children: [childId] }
			});
			else acc[id] = merge(acc[id] || {}, {
				type: "group",
				id,
				name: names[idx],
				tags: [],
				parent: paths[idx - 1],
				depth: idx,
				renderLabel,
				...childId && { children: [childId] }
			});
		});
		acc[item.id] = {
			type: "story",
			tags: [],
			...item,
			depth: paths.length,
			parent: paths[paths.length - 1],
			renderLabel,
			prepared: !!item.parameters
		};
		return acc;
	}, {});
	function addItem(acc, item) {
		if (acc[item.id]) return acc;
		acc[item.id] = item;
		if (item.type === "root" || item.type === "group" || item.type === "component") {
			item.children.forEach((childId) => addItem(acc, storiesHashOutOfOrder[childId]));
			item.tags = item.children.reduce((currentTags, childId) => {
				const child = acc[childId];
				return currentTags === null ? child.tags : intersect(currentTags, child.tags);
			}, null);
		}
		return acc;
	}
	const orphanHash = Object.values(storiesHashOutOfOrder).filter((i) => i.type !== "root" && !i.parent).reduce(addItem, {});
	return Object.values(storiesHashOutOfOrder).filter((i) => i.type === "root").reduce(addItem, orphanHash);
};
const transformStoryIndexV2toV3 = (index) => {
	return {
		v: 3,
		stories: Object.values(index.stories).reduce((acc, entry) => {
			acc[entry.id] = {
				...entry,
				title: entry.kind,
				name: entry.name || entry.story,
				importPath: entry.parameters.fileName || ""
			};
			return acc;
		}, {})
	};
};
const transformStoryIndexV3toV4 = (index) => {
	const countByTitle = (0, es_toolkit_array.countBy)(Object.values(index.stories), (item) => item.title);
	return {
		v: 4,
		entries: Object.values(index.stories).reduce((acc, entry) => {
			let type = "story";
			if (entry.parameters?.docsOnly || entry.name === "Page" && countByTitle[entry.title] === 1) type = "docs";
			acc[entry.id] = {
				type,
				...type === "docs" && {
					tags: ["stories-mdx"],
					storiesImports: []
				},
				...entry
			};
			delete acc[entry.id].story;
			delete acc[entry.id].kind;
			return acc;
		}, {})
	};
};
const transformStoryIndexV4toV5 = (index) => {
	return {
		v: 5,
		entries: Object.values(index.entries).reduce((acc, entry) => {
			acc[entry.id] = {
				...entry,
				tags: entry.tags ? [
					"dev",
					"test",
					...entry.tags
				] : ["dev"]
			};
			return acc;
		}, {})
	};
};
//#endregion
//#region src/util/tree.ts
const createId = (itemId, refId) => !refId || refId === "storybook_internal" ? itemId : `${refId}_${itemId}`;
const prevent = (e) => {
	e.preventDefault();
	return false;
};
const get = (0, memoizerific.default)(1e3)((id, dataset) => dataset[id]);
const getParent = (0, memoizerific.default)(1e3)((id, dataset) => {
	const item = get(id, dataset);
	return item && item.type !== "root" ? get(item.parent, dataset) : void 0;
});
const getParents = (0, memoizerific.default)(1e3)((id, dataset) => {
	const parent = getParent(id, dataset);
	return parent ? [parent, ...getParents(parent.id, dataset)] : [];
});
const getAncestorIds = (0, memoizerific.default)(1e3)((data, id) => getParents(id, data).map((item) => item.id));
const getDescendantIds = (0, memoizerific.default)(1e3)((data, id, skipLeafs) => {
	const entry = data[id];
	return (entry.type === "story" || entry.type === "docs" ? [] : entry.children).reduce((acc, childId) => {
		const child = data[childId];
		if (!child || skipLeafs && (child.type === "story" || child.type === "docs")) return acc;
		acc.push(childId, ...getDescendantIds(data, childId, skipLeafs));
		return acc;
	}, []);
});
function getPath(item, ref) {
	const parent = item.type !== "root" && item.parent ? ref.index[item.parent] : null;
	if (parent) return [...getPath(parent, ref), parent.name];
	return ref.id === "storybook_internal" ? [] : [ref.title || ref.id];
}
const searchItem = (item, ref) => {
	return {
		...item,
		refId: ref.id,
		path: getPath(item, ref)
	};
};
function cycle(array, index, delta) {
	let next = index + delta % array.length;
	if (next < 0) next = array.length + next;
	if (next >= array.length) next -= array.length;
	return next;
}
const getStateType = (isLoading, isAuthRequired, isError, isEmpty) => {
	switch (true) {
		case isAuthRequired: return "auth";
		case isError: return "error";
		case isLoading: return "loading";
		case isEmpty: return "empty";
		default: return "ready";
	}
};
const isAncestor = (element, maybeAncestor) => {
	if (!element || !maybeAncestor) return false;
	if (element === maybeAncestor) return true;
	return isAncestor(element.parentElement, maybeAncestor);
};
const removeNoiseFromName = (storyName) => storyName.replaceAll(/(\s|-|_)/gi, "");
const isStoryHoistable = (storyName, componentName) => removeNoiseFromName(storyName) === removeNoiseFromName(componentName);
//#endregion
//#region src/util/useStyle.ts
/**
* A hook to memoize a style. Uses `ViewStyle` per default, but can be used with other styles deriving from `FlexStyle` as well, such as `TextStyle`.
* @param styleFactory The function that returns a style
* @param deps The dependencies to trigger memoization re-evaluation
* @see ["Memoize!!! 💾 - a react (native) performance guide"](https://gist.github.com/mrousavy/0de7486814c655de8a110df5cef74ddc)
* @example
*
* // simple object styles
* const style1 = useStyle(() => ({ height: someDynamicValue }), [someDynamicValue])
*
* // array styles
* const style2 = useStyle(
*   () => [styles.container, props.style, { height: someDynamicValue }],
*   [props.style, someDynamicValue]
* );
*/
const useStyle = (styleFactory, deps) => (0, react.useMemo)(styleFactory, deps);
//#endregion
//#region src/util/string.ts
const CASE_SPLIT_PATTERN = /\p{Lu}?\p{Ll}+|[0-9]+|\p{Lu}+(?!\p{Ll})|\p{Emoji_Presentation}|\p{Extended_Pictographic}|\p{L}+/gu;
function words(str) {
	return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []);
}
function startCase(str) {
	const words$1 = words(str.trim());
	let result = "";
	for (let i = 0; i < words$1.length; i++) {
		const word = words$1[i];
		if (result) result += " ";
		result += word[0].toUpperCase() + word.slice(1).toLowerCase();
	}
	return result;
}
//#endregion
//#region src/hooks/useExpanded.ts
const initializeExpanded = ({ initialExpanded, rootIds }) => {
	const highlightedAncestors = [];
	const expandedState = initialExpanded;
	return [...rootIds, ...highlightedAncestors].reduce((acc, id) => Object.assign(acc, { [id]: id in expandedState ? expandedState[id] : true }), {});
};
const useExpanded = ({ refId, data, initialExpanded, rootIds, selectedStoryId }) => {
	const [expanded, setExpanded] = (0, react.useReducer)((state, { ids, value }) => ids.reduce((acc, id) => Object.assign(acc, { [id]: value }), { ...state }), {
		refId,
		data,
		rootIds,
		initialExpanded
	}, initializeExpanded);
	const updateExpanded = (0, react.useCallback)(({ ids, value }) => {
		setExpanded({
			ids,
			value
		});
	}, []);
	(0, react.useEffect)(() => {
		setExpanded({
			ids: getAncestorIds(data, selectedStoryId),
			value: true
		});
	}, [data, selectedStoryId]);
	return [expanded, updateExpanded];
};
//#endregion
//#region src/hooks/useLastViewed.ts
const useLastViewed = (selection) => {
	const lastViewedRef = (0, react.useRef)([]);
	const updateLastViewed = (0, react.useCallback)((story) => {
		const items = lastViewedRef.current;
		const index = items.findIndex(({ storyId, refId }) => storyId === story.storyId && refId === story.refId);
		if (index === 0) return;
		if (index === -1) lastViewedRef.current = [story, ...items];
		else lastViewedRef.current = [
			story,
			...items.slice(0, index),
			...items.slice(index + 1)
		];
	}, [lastViewedRef]);
	(0, react.useEffect)(() => {
		if (selection) updateLastViewed(selection);
	}, [selection]);
	return {
		getLastViewed: (0, react.useCallback)(() => lastViewedRef.current, [lastViewedRef]),
		clearLastViewed: (0, react.useCallback)(() => {
			lastViewedRef.current = lastViewedRef.current.slice(0, 1);
		}, [lastViewedRef])
	};
};
//#endregion
//#region src/hooks/useStoreState.ts
const useStoreBooleanState = (key, defaultValue) => {
	const storage = useStorage();
	const [val, setVal] = (0, react.useState)(defaultValue);
	(0, react.useEffect)(() => {
		Promise.resolve(storage.getItem(key)).then((newVal) => {
			if (newVal === null || newVal === void 0) setVal(defaultValue);
			else setVal(newVal === "true");
		});
	}, [
		key,
		storage,
		defaultValue
	]);
	(0, react.useEffect)(() => {
		storage.setItem(key, val.toString());
	}, [
		key,
		storage,
		val
	]);
	return [val, setVal];
};
const useStoreNumberState = (key, defaultValue) => {
	const storage = useStorage();
	const [val, setVal] = (0, react.useState)(defaultValue);
	(0, react.useEffect)(() => {
		Promise.resolve(storage.getItem(key)).then((newVal) => {
			if (newVal === null || newVal === void 0) setVal(defaultValue);
			else {
				const parsed = parseFloat(newVal);
				setVal(isNaN(parsed) ? defaultValue : parsed);
			}
		});
	}, [
		key,
		storage,
		defaultValue
	]);
	return [val, (0, react.useCallback)((value) => {
		setVal((prev) => {
			const next = typeof value === "function" ? value(prev) : value;
			storage.setItem(key, next.toString());
			return next;
		});
	}, [key, storage])];
};
//#endregion
exports.Button = Button;
exports.ButtonIcon = ButtonIcon;
exports.ButtonText = ButtonText;
exports.IconButton = IconButton;
exports.LayoutProvider = LayoutProvider;
exports.StorageProvider = StorageProvider;
exports.createId = createId;
exports.cycle = cycle;
exports.get = get;
exports.getAncestorIds = getAncestorIds;
exports.getDescendantIds = getDescendantIds;
exports.getParent = getParent;
exports.getParents = getParents;
exports.getPath = getPath;
exports.getStateType = getStateType;
exports.intersect = intersect;
exports.isAncestor = isAncestor;
exports.isExpandType = isExpandType;
exports.isStoryHoistable = isStoryHoistable;
exports.merge = merge;
exports.noArrayMerge = noArrayMerge;
exports.prevent = prevent;
exports.removeNoiseFromName = removeNoiseFromName;
exports.searchItem = searchItem;
exports.startCase = startCase;
exports.transformStoryIndexToStoriesHash = transformStoryIndexToStoriesHash;
exports.transformStoryIndexV2toV3 = transformStoryIndexV2toV3;
exports.transformStoryIndexV3toV4 = transformStoryIndexV3toV4;
exports.transformStoryIndexV4toV5 = transformStoryIndexV4toV5;
exports.useExpanded = useExpanded;
exports.useLastViewed = useLastViewed;
exports.useLayout = useLayout;
exports.useStorage = useStorage;
exports.useStoreBooleanState = useStoreBooleanState;
exports.useStoreNumberState = useStoreNumberState;
exports.useStyle = useStyle;
