// @storybook/react-native-web-vite@10.4.6/dist/preset.js
import CJS_COMPAT_NODE_URL_fhytmgh9zm from 'node:url';
import CJS_COMPAT_NODE_PATH_fhytmgh9zm from 'node:path';
import CJS_COMPAT_NODE_MODULE_fhytmgh9zm from "node:module";

var __filename = CJS_COMPAT_NODE_URL_fhytmgh9zm.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_fhytmgh9zm.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_fhytmgh9zm.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------

// src/preset.ts
import { viteFinal as reactViteFinal } from "@storybook/react-vite/preset";
import { rnw } from "vite-plugin-rnw";
import tsconfigPaths from "vite-tsconfig-paths";
var getExcludeOptions = (modulesToTranspile) => {
  let defaultModulesToTranspile = ["react-native", "@react-native", "expo", "@expo"], uniqueModulesToTranspile = Array.from(
    /* @__PURE__ */ new Set([...modulesToTranspile, ...defaultModulesToTranspile])
  );
  return modulesToTranspile.length ? { exclude: new RegExp(`/node_modules/(?!${uniqueModulesToTranspile.join("|")})`) } : {};
}, viteFinal = async (config, options) => {
  let { mergeConfig } = await import("vite"), { pluginReactOptions = {}, modulesToTranspile = [] } = await options.presets.apply("frameworkOptions"), { plugins = [], ...reactConfigWithoutPlugins } = await reactViteFinal(config, options);
  return mergeConfig(reactConfigWithoutPlugins, {
    plugins: [
      tsconfigPaths(),
      rnw({
        ...getExcludeOptions(modulesToTranspile),
        ...pluginReactOptions,
        jsxRuntime: pluginReactOptions.jsxRuntime || "automatic",
        babel: {
          babelrc: !1,
          configFile: !1,
          ...pluginReactOptions.babel
        }
      }),
      ...plugins
    ]
  });
}, core = {
  builder: import.meta.resolve("@storybook/builder-vite"),
  renderer: import.meta.resolve("@storybook/react/preset")
};
export {
  core,
  viteFinal
};
