// @storybook/react-native-theming@10.4.7/dist/index.js
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJSMin = (cb, mod) => () => (mod || (cb((mod = { exports: {} }).exports, mod), cb = null), mod.exports);
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
//#endregion
let react_native = require("react-native");
react_native = __toESM(react_native);
let react = require("react");
react = __toESM(react);
let polished = require("polished");
//#region ../../node_modules/postcss-value-parser/lib/parse.js
var require_parse = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	var openParentheses = "(".charCodeAt(0);
	var closeParentheses = ")".charCodeAt(0);
	var singleQuote = "'".charCodeAt(0);
	var doubleQuote = "\"".charCodeAt(0);
	var backslash = "\\".charCodeAt(0);
	var slash = "/".charCodeAt(0);
	var comma = ",".charCodeAt(0);
	var colon = ":".charCodeAt(0);
	var star = "*".charCodeAt(0);
	var uLower = "u".charCodeAt(0);
	var uUpper = "U".charCodeAt(0);
	var plus = "+".charCodeAt(0);
	var isUnicodeRange = /^[a-f0-9?-]+$/i;
	module.exports = function(input) {
		var tokens = [];
		var value = input;
		var next, quote, prev, token, escape, escapePos, whitespacePos, parenthesesOpenPos;
		var pos = 0;
		var code = value.charCodeAt(pos);
		var max = value.length;
		var stack = [{ nodes: tokens }];
		var balanced = 0;
		var parent;
		var name = "";
		var before = "";
		var after = "";
		while (pos < max) if (code <= 32) {
			next = pos;
			do {
				next += 1;
				code = value.charCodeAt(next);
			} while (code <= 32);
			token = value.slice(pos, next);
			prev = tokens[tokens.length - 1];
			if (code === closeParentheses && balanced) after = token;
			else if (prev && prev.type === "div") {
				prev.after = token;
				prev.sourceEndIndex += token.length;
			} else if (code === comma || code === colon || code === slash && value.charCodeAt(next + 1) !== star && (!parent || parent && parent.type === "function" && parent.value !== "calc")) before = token;
			else tokens.push({
				type: "space",
				sourceIndex: pos,
				sourceEndIndex: next,
				value: token
			});
			pos = next;
		} else if (code === singleQuote || code === doubleQuote) {
			next = pos;
			quote = code === singleQuote ? "'" : "\"";
			token = {
				type: "string",
				sourceIndex: pos,
				quote
			};
			do {
				escape = false;
				next = value.indexOf(quote, next + 1);
				if (~next) {
					escapePos = next;
					while (value.charCodeAt(escapePos - 1) === backslash) {
						escapePos -= 1;
						escape = !escape;
					}
				} else {
					value += quote;
					next = value.length - 1;
					token.unclosed = true;
				}
			} while (escape);
			token.value = value.slice(pos + 1, next);
			token.sourceEndIndex = token.unclosed ? next : next + 1;
			tokens.push(token);
			pos = next + 1;
			code = value.charCodeAt(pos);
		} else if (code === slash && value.charCodeAt(pos + 1) === star) {
			next = value.indexOf("*/", pos);
			token = {
				type: "comment",
				sourceIndex: pos,
				sourceEndIndex: next + 2
			};
			if (next === -1) {
				token.unclosed = true;
				next = value.length;
				token.sourceEndIndex = next;
			}
			token.value = value.slice(pos + 2, next);
			tokens.push(token);
			pos = next + 2;
			code = value.charCodeAt(pos);
		} else if ((code === slash || code === star) && parent && parent.type === "function" && parent.value === "calc") {
			token = value[pos];
			tokens.push({
				type: "word",
				sourceIndex: pos - before.length,
				sourceEndIndex: pos + token.length,
				value: token
			});
			pos += 1;
			code = value.charCodeAt(pos);
		} else if (code === slash || code === comma || code === colon) {
			token = value[pos];
			tokens.push({
				type: "div",
				sourceIndex: pos - before.length,
				sourceEndIndex: pos + token.length,
				value: token,
				before,
				after: ""
			});
			before = "";
			pos += 1;
			code = value.charCodeAt(pos);
		} else if (openParentheses === code) {
			next = pos;
			do {
				next += 1;
				code = value.charCodeAt(next);
			} while (code <= 32);
			parenthesesOpenPos = pos;
			token = {
				type: "function",
				sourceIndex: pos - name.length,
				value: name,
				before: value.slice(parenthesesOpenPos + 1, next)
			};
			pos = next;
			if (name === "url" && code !== singleQuote && code !== doubleQuote) {
				next -= 1;
				do {
					escape = false;
					next = value.indexOf(")", next + 1);
					if (~next) {
						escapePos = next;
						while (value.charCodeAt(escapePos - 1) === backslash) {
							escapePos -= 1;
							escape = !escape;
						}
					} else {
						value += ")";
						next = value.length - 1;
						token.unclosed = true;
					}
				} while (escape);
				whitespacePos = next;
				do {
					whitespacePos -= 1;
					code = value.charCodeAt(whitespacePos);
				} while (code <= 32);
				if (parenthesesOpenPos < whitespacePos) {
					if (pos !== whitespacePos + 1) token.nodes = [{
						type: "word",
						sourceIndex: pos,
						sourceEndIndex: whitespacePos + 1,
						value: value.slice(pos, whitespacePos + 1)
					}];
					else token.nodes = [];
					if (token.unclosed && whitespacePos + 1 !== next) {
						token.after = "";
						token.nodes.push({
							type: "space",
							sourceIndex: whitespacePos + 1,
							sourceEndIndex: next,
							value: value.slice(whitespacePos + 1, next)
						});
					} else {
						token.after = value.slice(whitespacePos + 1, next);
						token.sourceEndIndex = next;
					}
				} else {
					token.after = "";
					token.nodes = [];
				}
				pos = next + 1;
				token.sourceEndIndex = token.unclosed ? next : pos;
				code = value.charCodeAt(pos);
				tokens.push(token);
			} else {
				balanced += 1;
				token.after = "";
				token.sourceEndIndex = pos + 1;
				tokens.push(token);
				stack.push(token);
				tokens = token.nodes = [];
				parent = token;
			}
			name = "";
		} else if (closeParentheses === code && balanced) {
			pos += 1;
			code = value.charCodeAt(pos);
			parent.after = after;
			parent.sourceEndIndex += after.length;
			after = "";
			balanced -= 1;
			stack[stack.length - 1].sourceEndIndex = pos;
			stack.pop();
			parent = stack[balanced];
			tokens = parent.nodes;
		} else {
			next = pos;
			do {
				if (code === backslash) next += 1;
				next += 1;
				code = value.charCodeAt(next);
			} while (next < max && !(code <= 32 || code === singleQuote || code === doubleQuote || code === comma || code === colon || code === slash || code === openParentheses || code === star && parent && parent.type === "function" && parent.value === "calc" || code === slash && parent.type === "function" && parent.value === "calc" || code === closeParentheses && balanced));
			token = value.slice(pos, next);
			if (openParentheses === code) name = token;
			else if ((uLower === token.charCodeAt(0) || uUpper === token.charCodeAt(0)) && plus === token.charCodeAt(1) && isUnicodeRange.test(token.slice(2))) tokens.push({
				type: "unicode-range",
				sourceIndex: pos,
				sourceEndIndex: next,
				value: token
			});
			else tokens.push({
				type: "word",
				sourceIndex: pos,
				sourceEndIndex: next,
				value: token
			});
			pos = next;
		}
		for (pos = stack.length - 1; pos; pos -= 1) {
			stack[pos].unclosed = true;
			stack[pos].sourceEndIndex = value.length;
		}
		return stack[0].nodes;
	};
}));
//#endregion
//#region ../../node_modules/postcss-value-parser/lib/walk.js
var require_walk = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	module.exports = function walk(nodes, cb, bubble) {
		var i, max, node, result;
		for (i = 0, max = nodes.length; i < max; i += 1) {
			node = nodes[i];
			if (!bubble) result = cb(node, i, nodes);
			if (result !== false && node.type === "function" && Array.isArray(node.nodes)) walk(node.nodes, cb, bubble);
			if (bubble) cb(node, i, nodes);
		}
	};
}));
//#endregion
//#region ../../node_modules/postcss-value-parser/lib/stringify.js
var require_stringify = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	function stringifyNode(node, custom) {
		var type = node.type;
		var value = node.value;
		var buf;
		var customResult;
		if (custom && (customResult = custom(node)) !== void 0) return customResult;
		else if (type === "word" || type === "space") return value;
		else if (type === "string") {
			buf = node.quote || "";
			return buf + value + (node.unclosed ? "" : buf);
		} else if (type === "comment") return "/*" + value + (node.unclosed ? "" : "*/");
		else if (type === "div") return (node.before || "") + value + (node.after || "");
		else if (Array.isArray(node.nodes)) {
			buf = stringify(node.nodes, custom);
			if (type !== "function") return buf;
			return value + "(" + (node.before || "") + buf + (node.after || "") + (node.unclosed ? "" : ")");
		}
		return value;
	}
	function stringify(nodes, custom) {
		var result, i;
		if (Array.isArray(nodes)) {
			result = "";
			for (i = nodes.length - 1; ~i; i -= 1) result = stringifyNode(nodes[i], custom) + result;
			return result;
		}
		return stringifyNode(nodes, custom);
	}
	module.exports = stringify;
}));
//#endregion
//#region ../../node_modules/postcss-value-parser/lib/unit.js
var require_unit = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	var minus = "-".charCodeAt(0);
	var plus = "+".charCodeAt(0);
	var dot = ".".charCodeAt(0);
	var exp = "e".charCodeAt(0);
	var EXP = "E".charCodeAt(0);
	function likeNumber(value) {
		var code = value.charCodeAt(0);
		var nextCode;
		if (code === plus || code === minus) {
			nextCode = value.charCodeAt(1);
			if (nextCode >= 48 && nextCode <= 57) return true;
			var nextNextCode = value.charCodeAt(2);
			if (nextCode === dot && nextNextCode >= 48 && nextNextCode <= 57) return true;
			return false;
		}
		if (code === dot) {
			nextCode = value.charCodeAt(1);
			if (nextCode >= 48 && nextCode <= 57) return true;
			return false;
		}
		if (code >= 48 && code <= 57) return true;
		return false;
	}
	module.exports = function(value) {
		var pos = 0;
		var length = value.length;
		var code;
		var nextCode;
		var nextNextCode;
		if (length === 0 || !likeNumber(value)) return false;
		code = value.charCodeAt(pos);
		if (code === plus || code === minus) pos++;
		while (pos < length) {
			code = value.charCodeAt(pos);
			if (code < 48 || code > 57) break;
			pos += 1;
		}
		code = value.charCodeAt(pos);
		nextCode = value.charCodeAt(pos + 1);
		if (code === dot && nextCode >= 48 && nextCode <= 57) {
			pos += 2;
			while (pos < length) {
				code = value.charCodeAt(pos);
				if (code < 48 || code > 57) break;
				pos += 1;
			}
		}
		code = value.charCodeAt(pos);
		nextCode = value.charCodeAt(pos + 1);
		nextNextCode = value.charCodeAt(pos + 2);
		if ((code === exp || code === EXP) && (nextCode >= 48 && nextCode <= 57 || (nextCode === plus || nextCode === minus) && nextNextCode >= 48 && nextNextCode <= 57)) {
			pos += nextCode === plus || nextCode === minus ? 3 : 2;
			while (pos < length) {
				code = value.charCodeAt(pos);
				if (code < 48 || code > 57) break;
				pos += 1;
			}
		}
		return {
			number: value.slice(0, pos),
			unit: value.slice(pos)
		};
	};
}));
//#endregion
//#region ../../node_modules/postcss-value-parser/lib/index.js
var require_lib = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	var parse = require_parse();
	var walk = require_walk();
	var stringify = require_stringify();
	function ValueParser(value) {
		if (this instanceof ValueParser) {
			this.nodes = parse(value);
			return this;
		}
		return new ValueParser(value);
	}
	ValueParser.prototype.toString = function() {
		return Array.isArray(this.nodes) ? stringify(this.nodes) : "";
	};
	ValueParser.prototype.walk = function(cb, bubble) {
		walk(this.nodes, cb, bubble);
		return this;
	};
	ValueParser.unit = require_unit();
	ValueParser.walk = walk;
	ValueParser.stringify = stringify;
	module.exports = ValueParser;
}));
//#endregion
//#region ../../node_modules/camelize/index.js
var require_camelize = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	module.exports = function(obj) {
		if (typeof obj === "string") return camelCase(obj);
		return walk(obj);
	};
	function walk(obj) {
		if (!obj || typeof obj !== "object") return obj;
		if (isDate(obj) || isRegex(obj)) return obj;
		if (isArray(obj)) return map(obj, walk);
		return reduce(objectKeys(obj), function(acc, key) {
			var camel = camelCase(key);
			acc[camel] = walk(obj[key]);
			return acc;
		}, {});
	}
	function camelCase(str) {
		return str.replace(/[_.-](\w|$)/g, function(_, x) {
			return x.toUpperCase();
		});
	}
	var isArray = Array.isArray || function(obj) {
		return Object.prototype.toString.call(obj) === "[object Array]";
	};
	var isDate = function(obj) {
		return Object.prototype.toString.call(obj) === "[object Date]";
	};
	var isRegex = function(obj) {
		return Object.prototype.toString.call(obj) === "[object RegExp]";
	};
	var has = Object.prototype.hasOwnProperty;
	var objectKeys = Object.keys || function(obj) {
		var keys = [];
		for (var key in obj) if (has.call(obj, key)) keys.push(key);
		return keys;
	};
	function map(xs, f) {
		if (xs.map) return xs.map(f);
		var res = [];
		for (var i = 0; i < xs.length; i++) res.push(f(xs[i], i));
		return res;
	}
	function reduce(xs, f, acc) {
		if (xs.reduce) return xs.reduce(f, acc);
		for (var i = 0; i < xs.length; i++) acc = f(acc, xs[i], i);
		return acc;
	}
}));
//#endregion
//#region ../../node_modules/css-color-keywords/colors.json
var require_colors = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	module.exports = {
		"black": "#000000",
		"silver": "#c0c0c0",
		"gray": "#808080",
		"white": "#ffffff",
		"maroon": "#800000",
		"red": "#ff0000",
		"purple": "#800080",
		"fuchsia": "#ff00ff",
		"green": "#008000",
		"lime": "#00ff00",
		"olive": "#808000",
		"yellow": "#ffff00",
		"navy": "#000080",
		"blue": "#0000ff",
		"teal": "#008080",
		"aqua": "#00ffff",
		"orange": "#ffa500",
		"aliceblue": "#f0f8ff",
		"antiquewhite": "#faebd7",
		"aquamarine": "#7fffd4",
		"azure": "#f0ffff",
		"beige": "#f5f5dc",
		"bisque": "#ffe4c4",
		"blanchedalmond": "#ffebcd",
		"blueviolet": "#8a2be2",
		"brown": "#a52a2a",
		"burlywood": "#deb887",
		"cadetblue": "#5f9ea0",
		"chartreuse": "#7fff00",
		"chocolate": "#d2691e",
		"coral": "#ff7f50",
		"cornflowerblue": "#6495ed",
		"cornsilk": "#fff8dc",
		"crimson": "#dc143c",
		"darkblue": "#00008b",
		"darkcyan": "#008b8b",
		"darkgoldenrod": "#b8860b",
		"darkgray": "#a9a9a9",
		"darkgreen": "#006400",
		"darkgrey": "#a9a9a9",
		"darkkhaki": "#bdb76b",
		"darkmagenta": "#8b008b",
		"darkolivegreen": "#556b2f",
		"darkorange": "#ff8c00",
		"darkorchid": "#9932cc",
		"darkred": "#8b0000",
		"darksalmon": "#e9967a",
		"darkseagreen": "#8fbc8f",
		"darkslateblue": "#483d8b",
		"darkslategray": "#2f4f4f",
		"darkslategrey": "#2f4f4f",
		"darkturquoise": "#00ced1",
		"darkviolet": "#9400d3",
		"deeppink": "#ff1493",
		"deepskyblue": "#00bfff",
		"dimgray": "#696969",
		"dimgrey": "#696969",
		"dodgerblue": "#1e90ff",
		"firebrick": "#b22222",
		"floralwhite": "#fffaf0",
		"forestgreen": "#228b22",
		"gainsboro": "#dcdcdc",
		"ghostwhite": "#f8f8ff",
		"gold": "#ffd700",
		"goldenrod": "#daa520",
		"greenyellow": "#adff2f",
		"grey": "#808080",
		"honeydew": "#f0fff0",
		"hotpink": "#ff69b4",
		"indianred": "#cd5c5c",
		"indigo": "#4b0082",
		"ivory": "#fffff0",
		"khaki": "#f0e68c",
		"lavender": "#e6e6fa",
		"lavenderblush": "#fff0f5",
		"lawngreen": "#7cfc00",
		"lemonchiffon": "#fffacd",
		"lightblue": "#add8e6",
		"lightcoral": "#f08080",
		"lightcyan": "#e0ffff",
		"lightgoldenrodyellow": "#fafad2",
		"lightgray": "#d3d3d3",
		"lightgreen": "#90ee90",
		"lightgrey": "#d3d3d3",
		"lightpink": "#ffb6c1",
		"lightsalmon": "#ffa07a",
		"lightseagreen": "#20b2aa",
		"lightskyblue": "#87cefa",
		"lightslategray": "#778899",
		"lightslategrey": "#778899",
		"lightsteelblue": "#b0c4de",
		"lightyellow": "#ffffe0",
		"limegreen": "#32cd32",
		"linen": "#faf0e6",
		"mediumaquamarine": "#66cdaa",
		"mediumblue": "#0000cd",
		"mediumorchid": "#ba55d3",
		"mediumpurple": "#9370db",
		"mediumseagreen": "#3cb371",
		"mediumslateblue": "#7b68ee",
		"mediumspringgreen": "#00fa9a",
		"mediumturquoise": "#48d1cc",
		"mediumvioletred": "#c71585",
		"midnightblue": "#191970",
		"mintcream": "#f5fffa",
		"mistyrose": "#ffe4e1",
		"moccasin": "#ffe4b5",
		"navajowhite": "#ffdead",
		"oldlace": "#fdf5e6",
		"olivedrab": "#6b8e23",
		"orangered": "#ff4500",
		"orchid": "#da70d6",
		"palegoldenrod": "#eee8aa",
		"palegreen": "#98fb98",
		"paleturquoise": "#afeeee",
		"palevioletred": "#db7093",
		"papayawhip": "#ffefd5",
		"peachpuff": "#ffdab9",
		"peru": "#cd853f",
		"pink": "#ffc0cb",
		"plum": "#dda0dd",
		"powderblue": "#b0e0e6",
		"rosybrown": "#bc8f8f",
		"royalblue": "#4169e1",
		"saddlebrown": "#8b4513",
		"salmon": "#fa8072",
		"sandybrown": "#f4a460",
		"seagreen": "#2e8b57",
		"seashell": "#fff5ee",
		"sienna": "#a0522d",
		"skyblue": "#87ceeb",
		"slateblue": "#6a5acd",
		"slategray": "#708090",
		"slategrey": "#708090",
		"snow": "#fffafa",
		"springgreen": "#00ff7f",
		"steelblue": "#4682b4",
		"tan": "#d2b48c",
		"thistle": "#d8bfd8",
		"tomato": "#ff6347",
		"turquoise": "#40e0d0",
		"violet": "#ee82ee",
		"wheat": "#f5deb3",
		"whitesmoke": "#f5f5f5",
		"yellowgreen": "#9acd32",
		"rebeccapurple": "#663399"
	};
}));
//#endregion
//#region ../../node_modules/css-color-keywords/index.js
var require_css_color_keywords = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	module.exports = require_colors();
}));
//#endregion
//#region ../../node_modules/css-to-react-native/index.js
var require_css_to_react_native = /* @__PURE__ */ __commonJSMin(((exports) => {
	Object.defineProperty(exports, "__esModule", { value: true });
	function _interopDefault(ex) {
		return ex && typeof ex === "object" && "default" in ex ? ex["default"] : ex;
	}
	var parse = require_lib();
	var parse__default = _interopDefault(parse);
	var camelizeStyleName = _interopDefault(require_camelize());
	var cssColorKeywords = _interopDefault(require_css_color_keywords());
	var matchString = function matchString(node) {
		if (node.type !== "string") return null;
		return node.value.replace(/\\([0-9a-f]{1,6})(?:\s|$)/gi, function(match, charCode) {
			return String.fromCharCode(parseInt(charCode, 16));
		}).replace(/\\/g, "");
	};
	var hexColorRe = /^(#(?:[0-9a-f]{3,4}){1,2})$/i;
	var cssFunctionNameRe = /^(rgba?|hsla?|hwb|lab|lch|gray|color)$/;
	var matchColor = function matchColor(node) {
		if (node.type === "word" && (hexColorRe.test(node.value) || node.value in cssColorKeywords || node.value === "transparent")) return node.value;
		else if (node.type === "function" && cssFunctionNameRe.test(node.value)) return parse.stringify(node);
		return null;
	};
	var noneRe = /^(none)$/i;
	var autoRe = /^(auto)$/i;
	var identRe = /(^-?[_a-z][_a-z0-9-]*$)/i;
	var numberRe = /^([+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?)$/i;
	var lengthRe = /^(0$|(?:[+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?)(?=px$))/i;
	var unsupportedUnitRe = /^([+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?(ch|em|ex|rem|vh|vw|vmin|vmax|cm|mm|in|pc|pt))$/i;
	var angleRe = /^([+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?(?:deg|rad))$/i;
	var percentRe = /^([+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?%)$/i;
	var noopToken = function noopToken(predicate) {
		return function(node) {
			return predicate(node) ? "<token>" : null;
		};
	};
	var valueForTypeToken = function valueForTypeToken(type) {
		return function(node) {
			return node.type === type ? node.value : null;
		};
	};
	var regExpToken = function regExpToken(regExp, transform) {
		if (transform === void 0) transform = String;
		return function(node) {
			if (node.type !== "word") return null;
			var match = node.value.match(regExp);
			if (match === null) return null;
			return transform(match[1]);
		};
	};
	var SPACE = noopToken(function(node) {
		return node.type === "space";
	});
	var SLASH = noopToken(function(node) {
		return node.type === "div" && node.value === "/";
	});
	var COMMA = noopToken(function(node) {
		return node.type === "div" && node.value === ",";
	});
	var WORD = valueForTypeToken("word");
	var NONE = regExpToken(noneRe);
	var AUTO = regExpToken(autoRe);
	var NUMBER = regExpToken(numberRe, Number);
	var LENGTH = regExpToken(lengthRe, Number);
	var UNSUPPORTED_LENGTH_UNIT = regExpToken(unsupportedUnitRe);
	var ANGLE = regExpToken(angleRe, function(angle) {
		return angle.toLowerCase();
	});
	var PERCENT = regExpToken(percentRe);
	var IDENT = regExpToken(identRe);
	var STRING = matchString;
	var COLOR = matchColor;
	var LINE = regExpToken(/^(none|underline|line-through)$/i);
	var aspectRatio = function aspectRatio(tokenStream) {
		var aspectRatio = tokenStream.expect(NUMBER);
		if (tokenStream.hasTokens()) {
			tokenStream.expect(SLASH);
			aspectRatio /= tokenStream.expect(NUMBER);
		}
		return { aspectRatio };
	};
	var BORDER_STYLE = regExpToken(/^(solid|dashed|dotted)$/);
	var defaultBorderWidth = 1;
	var defaultBorderColor = "black";
	var defaultBorderStyle = "solid";
	var border = function border(tokenStream) {
		var borderWidth;
		var borderColor;
		var borderStyle;
		if (tokenStream.matches(NONE)) {
			tokenStream.expectEmpty();
			return {
				borderWidth: 0,
				borderColor: "black",
				borderStyle: "solid"
			};
		}
		var partsParsed = 0;
		while (partsParsed < 3 && tokenStream.hasTokens()) {
			if (partsParsed !== 0) tokenStream.expect(SPACE);
			if (borderWidth === void 0 && tokenStream.matches(LENGTH, UNSUPPORTED_LENGTH_UNIT)) borderWidth = tokenStream.lastValue;
			else if (borderColor === void 0 && tokenStream.matches(COLOR)) borderColor = tokenStream.lastValue;
			else if (borderStyle === void 0 && tokenStream.matches(BORDER_STYLE)) borderStyle = tokenStream.lastValue;
			else tokenStream["throw"]();
			partsParsed += 1;
		}
		tokenStream.expectEmpty();
		if (borderWidth === void 0) borderWidth = defaultBorderWidth;
		if (borderColor === void 0) borderColor = defaultBorderColor;
		if (borderStyle === void 0) borderStyle = defaultBorderStyle;
		return {
			borderWidth,
			borderColor,
			borderStyle
		};
	};
	var directionFactory = function directionFactory(_ref) {
		var _ref$types = _ref.types, types = _ref$types === void 0 ? [
			LENGTH,
			UNSUPPORTED_LENGTH_UNIT,
			PERCENT
		] : _ref$types, _ref$directions = _ref.directions, directions = _ref$directions === void 0 ? [
			"Top",
			"Right",
			"Bottom",
			"Left"
		] : _ref$directions, _ref$prefix = _ref.prefix, prefix = _ref$prefix === void 0 ? "" : _ref$prefix, _ref$suffix = _ref.suffix, suffix = _ref$suffix === void 0 ? "" : _ref$suffix;
		return function(tokenStream) {
			var _ref2;
			var values = [];
			values.push(tokenStream.expect.apply(tokenStream, types));
			while (values.length < 4 && tokenStream.hasTokens()) {
				tokenStream.expect(SPACE);
				values.push(tokenStream.expect.apply(tokenStream, types));
			}
			tokenStream.expectEmpty();
			var top = values[0], _values$ = values[1], right = _values$ === void 0 ? top : _values$, _values$2 = values[2], bottom = _values$2 === void 0 ? top : _values$2, _values$3 = values[3], left = _values$3 === void 0 ? right : _values$3;
			var keyFor = function keyFor(n) {
				return "" + prefix + directions[n] + suffix;
			};
			return _ref2 = {}, _ref2[keyFor(0)] = top, _ref2[keyFor(1)] = right, _ref2[keyFor(2)] = bottom, _ref2[keyFor(3)] = left, _ref2;
		};
	};
	var parseShadowOffset = function parseShadowOffset(tokenStream) {
		var width = tokenStream.expect(LENGTH);
		var height = tokenStream.matches(SPACE) ? tokenStream.expect(LENGTH) : width;
		tokenStream.expectEmpty();
		return {
			width,
			height
		};
	};
	var parseShadow = function parseShadow(tokenStream) {
		var offsetX;
		var offsetY;
		var radius;
		var color;
		if (tokenStream.matches(NONE)) {
			tokenStream.expectEmpty();
			return {
				offset: {
					width: 0,
					height: 0
				},
				radius: 0,
				color: "black"
			};
		}
		var didParseFirst = false;
		while (tokenStream.hasTokens()) {
			if (didParseFirst) tokenStream.expect(SPACE);
			if (offsetX === void 0 && tokenStream.matches(LENGTH, UNSUPPORTED_LENGTH_UNIT)) {
				offsetX = tokenStream.lastValue;
				tokenStream.expect(SPACE);
				offsetY = tokenStream.expect(LENGTH, UNSUPPORTED_LENGTH_UNIT);
				tokenStream.saveRewindPoint();
				if (tokenStream.matches(SPACE) && tokenStream.matches(LENGTH, UNSUPPORTED_LENGTH_UNIT)) radius = tokenStream.lastValue;
				else tokenStream.rewind();
			} else if (color === void 0 && tokenStream.matches(COLOR)) color = tokenStream.lastValue;
			else tokenStream["throw"]();
			didParseFirst = true;
		}
		if (offsetX === void 0) tokenStream["throw"]();
		return {
			offset: {
				width: offsetX,
				height: offsetY
			},
			radius: radius !== void 0 ? radius : 0,
			color: color !== void 0 ? color : "black"
		};
	};
	var boxShadow = function boxShadow(tokenStream) {
		var _parseShadow = parseShadow(tokenStream);
		return {
			shadowOffset: _parseShadow.offset,
			shadowRadius: _parseShadow.radius,
			shadowColor: _parseShadow.color,
			shadowOpacity: 1
		};
	};
	var defaultFlexGrow = 1;
	var defaultFlexShrink = 1;
	var defaultFlexBasis = 0;
	var flex = function flex(tokenStream) {
		var flexGrow;
		var flexShrink;
		var flexBasis;
		if (tokenStream.matches(NONE)) {
			tokenStream.expectEmpty();
			return {
				flexGrow: 0,
				flexShrink: 0,
				flexBasis: "auto"
			};
		}
		tokenStream.saveRewindPoint();
		if (tokenStream.matches(AUTO) && !tokenStream.hasTokens()) return {
			flexGrow: 1,
			flexShrink: 1,
			flexBasis: "auto"
		};
		tokenStream.rewind();
		var partsParsed = 0;
		while (partsParsed < 2 && tokenStream.hasTokens()) {
			if (partsParsed !== 0) tokenStream.expect(SPACE);
			if (flexGrow === void 0 && tokenStream.matches(NUMBER)) {
				flexGrow = tokenStream.lastValue;
				tokenStream.saveRewindPoint();
				if (tokenStream.matches(SPACE) && tokenStream.matches(NUMBER)) flexShrink = tokenStream.lastValue;
				else tokenStream.rewind();
			} else if (flexBasis === void 0 && tokenStream.matches(LENGTH, UNSUPPORTED_LENGTH_UNIT, PERCENT)) flexBasis = tokenStream.lastValue;
			else if (flexBasis === void 0 && tokenStream.matches(AUTO)) flexBasis = "auto";
			else tokenStream["throw"]();
			partsParsed += 1;
		}
		tokenStream.expectEmpty();
		if (flexGrow === void 0) flexGrow = defaultFlexGrow;
		if (flexShrink === void 0) flexShrink = defaultFlexShrink;
		if (flexBasis === void 0) flexBasis = defaultFlexBasis;
		return {
			flexGrow,
			flexShrink,
			flexBasis
		};
	};
	var FLEX_WRAP = regExpToken(/(nowrap|wrap|wrap-reverse)/);
	var FLEX_DIRECTION = regExpToken(/(row|row-reverse|column|column-reverse)/);
	var defaultFlexWrap = "nowrap";
	var defaultFlexDirection = "row";
	var flexFlow = function flexFlow(tokenStream) {
		var flexWrap;
		var flexDirection;
		var partsParsed = 0;
		while (partsParsed < 2 && tokenStream.hasTokens()) {
			if (partsParsed !== 0) tokenStream.expect(SPACE);
			if (flexWrap === void 0 && tokenStream.matches(FLEX_WRAP)) flexWrap = tokenStream.lastValue;
			else if (flexDirection === void 0 && tokenStream.matches(FLEX_DIRECTION)) flexDirection = tokenStream.lastValue;
			else tokenStream["throw"]();
			partsParsed += 1;
		}
		tokenStream.expectEmpty();
		if (flexWrap === void 0) flexWrap = defaultFlexWrap;
		if (flexDirection === void 0) flexDirection = defaultFlexDirection;
		return {
			flexWrap,
			flexDirection
		};
	};
	var fontFamily = function fontFamily(tokenStream) {
		var fontFamily;
		if (tokenStream.matches(STRING)) fontFamily = tokenStream.lastValue;
		else {
			fontFamily = tokenStream.expect(IDENT);
			while (tokenStream.hasTokens()) {
				tokenStream.expect(SPACE);
				var nextIdent = tokenStream.expect(IDENT);
				fontFamily += " " + nextIdent;
			}
		}
		tokenStream.expectEmpty();
		return { fontFamily };
	};
	var NORMAL = regExpToken(/^(normal)$/);
	var STYLE = regExpToken(/^(italic)$/);
	var WEIGHT = regExpToken(/^([1-9]00|bold)$/);
	var VARIANT = regExpToken(/^(small-caps)$/);
	var defaultFontStyle = "normal";
	var defaultFontWeight = "normal";
	var defaultFontVariant = [];
	var font = function font(tokenStream) {
		var fontStyle;
		var fontWeight;
		var fontVariant;
		var lineHeight;
		var numStyleWeightVariantMatched = 0;
		while (numStyleWeightVariantMatched < 3 && tokenStream.hasTokens()) {
			if (tokenStream.matches(NORMAL));
			else if (fontStyle === void 0 && tokenStream.matches(STYLE)) fontStyle = tokenStream.lastValue;
			else if (fontWeight === void 0 && tokenStream.matches(WEIGHT)) fontWeight = tokenStream.lastValue;
			else if (fontVariant === void 0 && tokenStream.matches(VARIANT)) fontVariant = [tokenStream.lastValue];
			else break;
			tokenStream.expect(SPACE);
			numStyleWeightVariantMatched += 1;
		}
		var fontSize = tokenStream.expect(LENGTH, UNSUPPORTED_LENGTH_UNIT);
		if (tokenStream.matches(SLASH)) lineHeight = tokenStream.expect(LENGTH, UNSUPPORTED_LENGTH_UNIT);
		tokenStream.expect(SPACE);
		var fontFamily$1 = fontFamily(tokenStream).fontFamily;
		if (fontStyle === void 0) fontStyle = defaultFontStyle;
		if (fontWeight === void 0) fontWeight = defaultFontWeight;
		if (fontVariant === void 0) fontVariant = defaultFontVariant;
		var out = {
			fontStyle,
			fontWeight,
			fontVariant,
			fontSize,
			fontFamily: fontFamily$1
		};
		if (lineHeight !== void 0) out.lineHeight = lineHeight;
		return out;
	};
	var fontVariant = function fontVariant(tokenStream) {
		var values = [tokenStream.expect(IDENT)];
		while (tokenStream.hasTokens()) {
			tokenStream.expect(SPACE);
			values.push(tokenStream.expect(IDENT));
		}
		return { fontVariant: values };
	};
	var ALIGN_CONTENT = regExpToken(/(flex-(?:start|end)|center|stretch|space-(?:between|around))/);
	var JUSTIFY_CONTENT = regExpToken(/(flex-(?:start|end)|center|space-(?:between|around|evenly))/);
	var placeContent = function placeContent(tokenStream) {
		var alignContent = tokenStream.expect(ALIGN_CONTENT);
		var justifyContent;
		if (tokenStream.hasTokens()) {
			tokenStream.expect(SPACE);
			justifyContent = tokenStream.expect(JUSTIFY_CONTENT);
		} else justifyContent = "stretch";
		tokenStream.expectEmpty();
		return {
			alignContent,
			justifyContent
		};
	};
	var STYLE$1 = regExpToken(/^(solid|double|dotted|dashed)$/);
	var defaultTextDecorationLine = "none";
	var defaultTextDecorationStyle = "solid";
	var defaultTextDecorationColor = "black";
	var textDecoration = function textDecoration(tokenStream) {
		var line;
		var style;
		var color;
		var didParseFirst = false;
		while (tokenStream.hasTokens()) {
			if (didParseFirst) tokenStream.expect(SPACE);
			if (line === void 0 && tokenStream.matches(LINE)) {
				var lines = [tokenStream.lastValue.toLowerCase()];
				tokenStream.saveRewindPoint();
				if (lines[0] !== "none" && tokenStream.matches(SPACE) && tokenStream.matches(LINE)) {
					lines.push(tokenStream.lastValue.toLowerCase());
					lines.sort().reverse();
				} else tokenStream.rewind();
				line = lines.join(" ");
			} else if (style === void 0 && tokenStream.matches(STYLE$1)) style = tokenStream.lastValue;
			else if (color === void 0 && tokenStream.matches(COLOR)) color = tokenStream.lastValue;
			else tokenStream["throw"]();
			didParseFirst = true;
		}
		return {
			textDecorationLine: line !== void 0 ? line : defaultTextDecorationLine,
			textDecorationColor: color !== void 0 ? color : defaultTextDecorationColor,
			textDecorationStyle: style !== void 0 ? style : defaultTextDecorationStyle
		};
	};
	var textDecorationLine = function textDecorationLine(tokenStream) {
		var lines = [];
		var didParseFirst = false;
		while (tokenStream.hasTokens()) {
			if (didParseFirst) tokenStream.expect(SPACE);
			lines.push(tokenStream.expect(LINE).toLowerCase());
			didParseFirst = true;
		}
		lines.sort().reverse();
		return { textDecorationLine: lines.join(" ") };
	};
	var textShadow = function textShadow(tokenStream) {
		var _parseShadow2 = parseShadow(tokenStream);
		return {
			textShadowOffset: _parseShadow2.offset,
			textShadowRadius: _parseShadow2.radius,
			textShadowColor: _parseShadow2.color
		};
	};
	var oneOfType = function oneOfType(tokenType) {
		return function(functionStream) {
			var value = functionStream.expect(tokenType);
			functionStream.expectEmpty();
			return value;
		};
	};
	var singleNumber = oneOfType(NUMBER);
	var singleLength = oneOfType(LENGTH);
	var singleAngle = oneOfType(ANGLE);
	var xyTransformFactory = function xyTransformFactory(tokenType) {
		return function(key, valueIfOmitted) {
			return function(functionStream) {
				var _ref3, _ref4;
				var x = functionStream.expect(tokenType);
				var y;
				if (functionStream.hasTokens()) {
					functionStream.expect(COMMA);
					y = functionStream.expect(tokenType);
				} else if (valueIfOmitted !== void 0) y = valueIfOmitted;
				else return x;
				functionStream.expectEmpty();
				return [(_ref3 = {}, _ref3[key + "Y"] = y, _ref3), (_ref4 = {}, _ref4[key + "X"] = x, _ref4)];
			};
		};
	};
	var xyNumber = xyTransformFactory(NUMBER);
	var xyLength = xyTransformFactory(LENGTH);
	var xyAngle = xyTransformFactory(ANGLE);
	var partTransforms = {
		perspective: singleNumber,
		scale: xyNumber("scale"),
		scaleX: singleNumber,
		scaleY: singleNumber,
		translate: xyLength("translate", 0),
		translateX: singleLength,
		translateY: singleLength,
		rotate: singleAngle,
		rotateX: singleAngle,
		rotateY: singleAngle,
		rotateZ: singleAngle,
		skewX: singleAngle,
		skewY: singleAngle,
		skew: xyAngle("skew", "0deg")
	};
	var transforms = {
		aspectRatio,
		background: function background(tokenStream) {
			return { backgroundColor: tokenStream.expect(COLOR) };
		},
		border,
		borderColor: directionFactory({
			types: [COLOR],
			prefix: "border",
			suffix: "Color"
		}),
		borderRadius: directionFactory({
			directions: [
				"TopLeft",
				"TopRight",
				"BottomRight",
				"BottomLeft"
			],
			prefix: "border",
			suffix: "Radius"
		}),
		borderWidth: directionFactory({
			prefix: "border",
			suffix: "Width"
		}),
		boxShadow,
		flex,
		flexFlow,
		font,
		fontFamily,
		fontVariant,
		fontWeight: function fontWeight(tokenStream) {
			return { fontWeight: tokenStream.expect(WORD) };
		},
		margin: directionFactory({
			types: [
				LENGTH,
				UNSUPPORTED_LENGTH_UNIT,
				PERCENT,
				AUTO
			],
			prefix: "margin"
		}),
		padding: directionFactory({ prefix: "padding" }),
		placeContent,
		shadowOffset: function shadowOffset(tokenStream) {
			return { shadowOffset: parseShadowOffset(tokenStream) };
		},
		textShadow,
		textShadowOffset: function textShadowOffset(tokenStream) {
			return { textShadowOffset: parseShadowOffset(tokenStream) };
		},
		textDecoration,
		textDecorationLine,
		transform: function transform(tokenStream) {
			var transforms = [];
			var didParseFirst = false;
			while (tokenStream.hasTokens()) {
				if (didParseFirst) tokenStream.expect(SPACE);
				var functionStream = tokenStream.expectFunction();
				var functionName = functionStream.functionName;
				var transformedValues = partTransforms[functionName](functionStream);
				if (!Array.isArray(transformedValues)) {
					var _ref5;
					transformedValues = [(_ref5 = {}, _ref5[functionName] = transformedValues, _ref5)];
				}
				transforms = transformedValues.concat(transforms);
				didParseFirst = true;
			}
			return { transform: transforms };
		}
	};
	var propertiesWithoutUnits;
	if (process.env.NODE_ENV !== "production") propertiesWithoutUnits = [
		"aspectRatio",
		"elevation",
		"flexGrow",
		"flexShrink",
		"opacity",
		"shadowOpacity",
		"zIndex"
	];
	var devPropertiesWithUnitsRegExp = propertiesWithoutUnits != null ? new RegExp(propertiesWithoutUnits.join("|")) : null;
	var SYMBOL_MATCH = "SYMBOL_MATCH";
	var TokenStream = /*#__PURE__*/ function() {
		function TokenStream(nodes, parent) {
			this.index = 0;
			this.nodes = nodes;
			this.functionName = parent != null ? parent.value : null;
			this.lastValue = null;
			this.rewindIndex = -1;
		}
		var _proto = TokenStream.prototype;
		_proto.hasTokens = function hasTokens() {
			return this.index <= this.nodes.length - 1;
		};
		_proto[SYMBOL_MATCH] = function() {
			if (!this.hasTokens()) return null;
			var node = this.nodes[this.index];
			for (var i = 0; i < arguments.length; i += 1) {
				var value = (i < 0 || arguments.length <= i ? void 0 : arguments[i])(node);
				if (value !== null) {
					this.index += 1;
					this.lastValue = value;
					return value;
				}
			}
			return null;
		};
		_proto.matches = function matches() {
			return this[SYMBOL_MATCH].apply(this, arguments) !== null;
		};
		_proto.expect = function expect() {
			var value = this[SYMBOL_MATCH].apply(this, arguments);
			return value !== null ? value : this["throw"]();
		};
		_proto.matchesFunction = function matchesFunction() {
			var node = this.nodes[this.index];
			if (node.type !== "function") return null;
			var value = new TokenStream(node.nodes, node);
			this.index += 1;
			this.lastValue = null;
			return value;
		};
		_proto.expectFunction = function expectFunction() {
			var value = this.matchesFunction();
			return value !== null ? value : this["throw"]();
		};
		_proto.expectEmpty = function expectEmpty() {
			if (this.hasTokens()) this["throw"]();
		};
		_proto["throw"] = function _throw() {
			throw new Error("Unexpected token type: " + this.nodes[this.index].type);
		};
		_proto.saveRewindPoint = function saveRewindPoint() {
			this.rewindIndex = this.index;
		};
		_proto.rewind = function rewind() {
			if (this.rewindIndex === -1) throw new Error("Internal error");
			this.index = this.rewindIndex;
			this.lastValue = null;
		};
		return TokenStream;
	}();
	var numberOrLengthRe = /^([+-]?(?:\d*\.)?\d+(?:e[+-]?\d+)?)(?:px)?$/i;
	var numberOnlyRe = /^[+-]?(?:\d*\.\d*|[1-9]\d*)(?:e[+-]?\d+)?$/i;
	var boolRe = /^true|false$/i;
	var nullRe = /^null$/i;
	var undefinedRe = /^undefined$/i;
	var transformRawValue = function transformRawValue(propName, value) {
		if (process.env.NODE_ENV !== "production") {
			var needsUnit = !devPropertiesWithUnitsRegExp.test(propName);
			var isNumberWithoutUnit = numberOnlyRe.test(value);
			if (needsUnit && isNumberWithoutUnit) console.warn("Expected style \"" + propName + ": " + value + "\" to contain units");
			if (!needsUnit && value !== "0" && !isNumberWithoutUnit) console.warn("Expected style \"" + propName + ": " + value + "\" to be unitless");
		}
		var numberMatch = value.match(numberOrLengthRe);
		if (numberMatch !== null) return Number(numberMatch[1]);
		var boolMatch = value.match(boolRe);
		if (boolMatch !== null) return boolMatch[0].toLowerCase() === "true";
		if (value.match(nullRe) !== null) return null;
		if (value.match(undefinedRe) !== null) return void 0;
		return value;
	};
	var baseTransformShorthandValue = function baseTransformShorthandValue(propName, value) {
		var tokenStream = new TokenStream(parse__default(value).nodes);
		return transforms[propName](tokenStream);
	};
	var transformShorthandValue = process.env.NODE_ENV === "production" ? baseTransformShorthandValue : function(propName, value) {
		try {
			return baseTransformShorthandValue(propName, value);
		} catch (e) {
			throw new Error("Failed to parse declaration \"" + propName + ": " + value + "\"");
		}
	};
	var getStylesForProperty = function getStylesForProperty(propName, inputValue, allowShorthand) {
		var _ref6;
		var isRawValue = allowShorthand === false || !(propName in transforms);
		var value = inputValue.trim();
		return isRawValue ? (_ref6 = {}, _ref6[propName] = transformRawValue(propName, value), _ref6) : transformShorthandValue(propName, value);
	};
	var getPropertyName = function getPropertyName(propName) {
		if (/^--\w+/.test(propName)) return propName;
		return camelizeStyleName(propName);
	};
	exports["default"] = function index(rules, shorthandBlacklist) {
		if (shorthandBlacklist === void 0) shorthandBlacklist = [];
		return rules.reduce(function(accum, rule) {
			var propertyName = getPropertyName(rule[0]);
			var value = rule[1];
			var allowShorthand = shorthandBlacklist.indexOf(propertyName) === -1;
			return Object.assign(accum, getStylesForProperty(propertyName, value, allowShorthand));
		}, {});
	};
	exports.getPropertyName = getPropertyName;
	exports.getStylesForProperty = getStylesForProperty;
	exports.transformRawValue = transformRawValue;
}));
//#endregion
//#region ../../node_modules/@emotion/sheet/dist/emotion-sheet.esm.js
function sheetForTag(tag) {
	if (tag.sheet) return tag.sheet;
	/* istanbul ignore next */
	for (var i = 0; i < document.styleSheets.length; i++) if (document.styleSheets[i].ownerNode === tag) return document.styleSheets[i];
}
function createStyleElement(options) {
	var tag = document.createElement("style");
	tag.setAttribute("data-emotion", options.key);
	if (options.nonce !== void 0) tag.setAttribute("nonce", options.nonce);
	tag.appendChild(document.createTextNode(""));
	tag.setAttribute("data-s", "");
	return tag;
}
var StyleSheet$1 = /*#__PURE__*/ function() {
	function StyleSheet(options) {
		var _this = this;
		this._insertTag = function(tag) {
			var before;
			if (_this.tags.length === 0) if (_this.insertionPoint) before = _this.insertionPoint.nextSibling;
			else if (_this.prepend) before = _this.container.firstChild;
			else before = _this.before;
			else before = _this.tags[_this.tags.length - 1].nextSibling;
			_this.container.insertBefore(tag, before);
			_this.tags.push(tag);
		};
		this.isSpeedy = options.speedy === void 0 ? true : options.speedy;
		this.tags = [];
		this.ctr = 0;
		this.nonce = options.nonce;
		this.key = options.key;
		this.container = options.container;
		this.prepend = options.prepend;
		this.insertionPoint = options.insertionPoint;
		this.before = null;
	}
	var _proto = StyleSheet.prototype;
	_proto.hydrate = function hydrate(nodes) {
		nodes.forEach(this._insertTag);
	};
	_proto.insert = function insert(rule) {
		if (this.ctr % (this.isSpeedy ? 65e3 : 1) === 0) this._insertTag(createStyleElement(this));
		var tag = this.tags[this.tags.length - 1];
		if (this.isSpeedy) {
			var sheet = sheetForTag(tag);
			try {
				sheet.insertRule(rule, sheet.cssRules.length);
			} catch (e) {}
		} else tag.appendChild(document.createTextNode(rule));
		this.ctr++;
	};
	_proto.flush = function flush() {
		this.tags.forEach(function(tag) {
			var _tag$parentNode;
			return (_tag$parentNode = tag.parentNode) == null ? void 0 : _tag$parentNode.removeChild(tag);
		});
		this.tags = [];
		this.ctr = 0;
	};
	return StyleSheet;
}();
//#endregion
//#region ../../node_modules/stylis/src/Enum.js
var MS = "-ms-";
var MOZ = "-moz-";
var WEBKIT = "-webkit-";
var COMMENT = "comm";
var RULESET = "rule";
var DECLARATION = "decl";
var IMPORT = "@import";
var KEYFRAMES = "@keyframes";
var LAYER = "@layer";
//#endregion
//#region ../../node_modules/stylis/src/Utility.js
/**
* @param {number}
* @return {number}
*/
var abs = Math.abs;
/**
* @param {number}
* @return {string}
*/
var from = String.fromCharCode;
/**
* @param {object}
* @return {object}
*/
var assign = Object.assign;
/**
* @param {string} value
* @param {number} length
* @return {number}
*/
function hash(value, length) {
	return charat(value, 0) ^ 45 ? (((length << 2 ^ charat(value, 0)) << 2 ^ charat(value, 1)) << 2 ^ charat(value, 2)) << 2 ^ charat(value, 3) : 0;
}
/**
* @param {string} value
* @return {string}
*/
function trim(value) {
	return value.trim();
}
/**
* @param {string} value
* @param {RegExp} pattern
* @return {string?}
*/
function match(value, pattern) {
	return (value = pattern.exec(value)) ? value[0] : value;
}
/**
* @param {string} value
* @param {(string|RegExp)} pattern
* @param {string} replacement
* @return {string}
*/
function replace(value, pattern, replacement) {
	return value.replace(pattern, replacement);
}
/**
* @param {string} value
* @param {string} search
* @return {number}
*/
function indexof(value, search) {
	return value.indexOf(search);
}
/**
* @param {string} value
* @param {number} index
* @return {number}
*/
function charat(value, index) {
	return value.charCodeAt(index) | 0;
}
/**
* @param {string} value
* @param {number} begin
* @param {number} end
* @return {string}
*/
function substr(value, begin, end) {
	return value.slice(begin, end);
}
/**
* @param {string} value
* @return {number}
*/
function strlen(value) {
	return value.length;
}
/**
* @param {any[]} value
* @return {number}
*/
function sizeof(value) {
	return value.length;
}
/**
* @param {any} value
* @param {any[]} array
* @return {any}
*/
function append(value, array) {
	return array.push(value), value;
}
/**
* @param {string[]} array
* @param {function} callback
* @return {string}
*/
function combine(array, callback) {
	return array.map(callback).join("");
}
//#endregion
//#region ../../node_modules/stylis/src/Tokenizer.js
var line = 1;
var column = 1;
var length = 0;
var position = 0;
var character = 0;
var characters = "";
/**
* @param {string} value
* @param {object | null} root
* @param {object | null} parent
* @param {string} type
* @param {string[] | string} props
* @param {object[] | string} children
* @param {number} length
*/
function node(value, root, parent, type, props, children, length) {
	return {
		value,
		root,
		parent,
		type,
		props,
		children,
		line,
		column,
		length,
		return: ""
	};
}
/**
* @param {object} root
* @param {object} props
* @return {object}
*/
function copy(root, props) {
	return assign(node("", null, null, "", null, null, 0), root, { length: -root.length }, props);
}
/**
* @return {number}
*/
function char() {
	return character;
}
/**
* @return {number}
*/
function prev() {
	character = position > 0 ? charat(characters, --position) : 0;
	if (column--, character === 10) column = 1, line--;
	return character;
}
/**
* @return {number}
*/
function next() {
	character = position < length ? charat(characters, position++) : 0;
	if (column++, character === 10) column = 1, line++;
	return character;
}
/**
* @return {number}
*/
function peek() {
	return charat(characters, position);
}
/**
* @return {number}
*/
function caret() {
	return position;
}
/**
* @param {number} begin
* @param {number} end
* @return {string}
*/
function slice(begin, end) {
	return substr(characters, begin, end);
}
/**
* @param {number} type
* @return {number}
*/
function token(type) {
	switch (type) {
		case 0:
		case 9:
		case 10:
		case 13:
		case 32: return 5;
		case 33:
		case 43:
		case 44:
		case 47:
		case 62:
		case 64:
		case 126:
		case 59:
		case 123:
		case 125: return 4;
		case 58: return 3;
		case 34:
		case 39:
		case 40:
		case 91: return 2;
		case 41:
		case 93: return 1;
	}
	return 0;
}
/**
* @param {string} value
* @return {any[]}
*/
function alloc(value) {
	return line = column = 1, length = strlen(characters = value), position = 0, [];
}
/**
* @param {any} value
* @return {any}
*/
function dealloc(value) {
	return characters = "", value;
}
/**
* @param {number} type
* @return {string}
*/
function delimit(type) {
	return trim(slice(position - 1, delimiter(type === 91 ? type + 2 : type === 40 ? type + 1 : type)));
}
/**
* @param {number} type
* @return {string}
*/
function whitespace(type) {
	while (character = peek()) if (character < 33) next();
	else break;
	return token(type) > 2 || token(character) > 3 ? "" : " ";
}
/**
* @param {number} index
* @param {number} count
* @return {string}
*/
function escaping(index, count) {
	while (--count && next()) if (character < 48 || character > 102 || character > 57 && character < 65 || character > 70 && character < 97) break;
	return slice(index, caret() + (count < 6 && peek() == 32 && next() == 32));
}
/**
* @param {number} type
* @return {number}
*/
function delimiter(type) {
	while (next()) switch (character) {
		case type: return position;
		case 34:
		case 39:
			if (type !== 34 && type !== 39) delimiter(character);
			break;
		case 40:
			if (type === 41) delimiter(type);
			break;
		case 92:
			next();
			break;
	}
	return position;
}
/**
* @param {number} type
* @param {number} index
* @return {number}
*/
function commenter(type, index) {
	while (next()) if (type + character === 57) break;
	else if (type + character === 84 && peek() === 47) break;
	return "/*" + slice(index, position - 1) + "*" + from(type === 47 ? type : next());
}
/**
* @param {number} index
* @return {string}
*/
function identifier(index) {
	while (!token(peek())) next();
	return slice(index, position);
}
//#endregion
//#region ../../node_modules/stylis/src/Parser.js
/**
* @param {string} value
* @return {object[]}
*/
function compile(value) {
	return dealloc(parse("", null, null, null, [""], value = alloc(value), 0, [0], value));
}
/**
* @param {string} value
* @param {object} root
* @param {object?} parent
* @param {string[]} rule
* @param {string[]} rules
* @param {string[]} rulesets
* @param {number[]} pseudo
* @param {number[]} points
* @param {string[]} declarations
* @return {object}
*/
function parse(value, root, parent, rule, rules, rulesets, pseudo, points, declarations) {
	var index = 0;
	var offset = 0;
	var length = pseudo;
	var atrule = 0;
	var property = 0;
	var previous = 0;
	var variable = 1;
	var scanning = 1;
	var ampersand = 1;
	var character = 0;
	var type = "";
	var props = rules;
	var children = rulesets;
	var reference = rule;
	var characters = type;
	while (scanning) switch (previous = character, character = next()) {
		case 40: if (previous != 108 && charat(characters, length - 1) == 58) {
			if (indexof(characters += replace(delimit(character), "&", "&\f"), "&\f") != -1) ampersand = -1;
			break;
		}
		case 34:
		case 39:
		case 91:
			characters += delimit(character);
			break;
		case 9:
		case 10:
		case 13:
		case 32:
			characters += whitespace(previous);
			break;
		case 92:
			characters += escaping(caret() - 1, 7);
			continue;
		case 47:
			switch (peek()) {
				case 42:
				case 47:
					append(comment(commenter(next(), caret()), root, parent), declarations);
					break;
				default: characters += "/";
			}
			break;
		case 123 * variable: points[index++] = strlen(characters) * ampersand;
		case 125 * variable:
		case 59:
		case 0:
			switch (character) {
				case 0:
				case 125: scanning = 0;
				case 59 + offset:
					if (ampersand == -1) characters = replace(characters, /\f/g, "");
					if (property > 0 && strlen(characters) - length) append(property > 32 ? declaration(characters + ";", rule, parent, length - 1) : declaration(replace(characters, " ", "") + ";", rule, parent, length - 2), declarations);
					break;
				case 59: characters += ";";
				default:
					append(reference = ruleset(characters, root, parent, index, offset, rules, points, type, props = [], children = [], length), rulesets);
					if (character === 123) if (offset === 0) parse(characters, root, reference, reference, props, rulesets, length, points, children);
					else switch (atrule === 99 && charat(characters, 3) === 110 ? 100 : atrule) {
						case 100:
						case 108:
						case 109:
						case 115:
							parse(value, reference, reference, rule && append(ruleset(value, reference, reference, 0, 0, rules, points, type, rules, props = [], length), children), rules, children, length, points, rule ? props : children);
							break;
						default: parse(characters, reference, reference, reference, [""], children, 0, points, children);
					}
			}
			index = offset = property = 0, variable = ampersand = 1, type = characters = "", length = pseudo;
			break;
		case 58: length = 1 + strlen(characters), property = previous;
		default:
			if (variable < 1) {
				if (character == 123) --variable;
				else if (character == 125 && variable++ == 0 && prev() == 125) continue;
			}
			switch (characters += from(character), character * variable) {
				case 38:
					ampersand = offset > 0 ? 1 : (characters += "\f", -1);
					break;
				case 44:
					points[index++] = (strlen(characters) - 1) * ampersand, ampersand = 1;
					break;
				case 64:
					if (peek() === 45) characters += delimit(next());
					atrule = peek(), offset = length = strlen(type = characters += identifier(caret())), character++;
					break;
				case 45: if (previous === 45 && strlen(characters) == 2) variable = 0;
			}
	}
	return rulesets;
}
/**
* @param {string} value
* @param {object} root
* @param {object?} parent
* @param {number} index
* @param {number} offset
* @param {string[]} rules
* @param {number[]} points
* @param {string} type
* @param {string[]} props
* @param {string[]} children
* @param {number} length
* @return {object}
*/
function ruleset(value, root, parent, index, offset, rules, points, type, props, children, length) {
	var post = offset - 1;
	var rule = offset === 0 ? rules : [""];
	var size = sizeof(rule);
	for (var i = 0, j = 0, k = 0; i < index; ++i) for (var x = 0, y = substr(value, post + 1, post = abs(j = points[i])), z = value; x < size; ++x) if (z = trim(j > 0 ? rule[x] + " " + y : replace(y, /&\f/g, rule[x]))) props[k++] = z;
	return node(value, root, parent, offset === 0 ? RULESET : type, props, children, length);
}
/**
* @param {number} value
* @param {object} root
* @param {object?} parent
* @return {object}
*/
function comment(value, root, parent) {
	return node(value, root, parent, COMMENT, from(char()), substr(value, 2, -2), 0);
}
/**
* @param {string} value
* @param {object} root
* @param {object?} parent
* @param {number} length
* @return {object}
*/
function declaration(value, root, parent, length) {
	return node(value, root, parent, DECLARATION, substr(value, 0, length), substr(value, length + 1, -1), length);
}
//#endregion
//#region ../../node_modules/stylis/src/Serializer.js
/**
* @param {object[]} children
* @param {function} callback
* @return {string}
*/
function serialize(children, callback) {
	var output = "";
	var length = sizeof(children);
	for (var i = 0; i < length; i++) output += callback(children[i], i, children, callback) || "";
	return output;
}
/**
* @param {object} element
* @param {number} index
* @param {object[]} children
* @param {function} callback
* @return {string}
*/
function stringify(element, index, children, callback) {
	switch (element.type) {
		case LAYER: if (element.children.length) break;
		case IMPORT:
		case DECLARATION: return element.return = element.return || element.value;
		case COMMENT: return "";
		case KEYFRAMES: return element.return = element.value + "{" + serialize(element.children, callback) + "}";
		case RULESET: element.value = element.props.join(",");
	}
	return strlen(children = serialize(element.children, callback)) ? element.return = element.value + "{" + children + "}" : "";
}
//#endregion
//#region ../../node_modules/stylis/src/Middleware.js
/**
* @param {function[]} collection
* @return {function}
*/
function middleware(collection) {
	var length = sizeof(collection);
	return function(element, index, children, callback) {
		var output = "";
		for (var i = 0; i < length; i++) output += collection[i](element, index, children, callback) || "";
		return output;
	};
}
/**
* @param {function} callback
* @return {function}
*/
function rulesheet(callback) {
	return function(element) {
		if (!element.root) {
			if (element = element.return) callback(element);
		}
	};
}
//#endregion
//#region ../../node_modules/@emotion/weak-memoize/dist/emotion-weak-memoize.esm.js
var weakMemoize = function weakMemoize(func) {
	var cache = /* @__PURE__ */ new WeakMap();
	return function(arg) {
		if (cache.has(arg)) return cache.get(arg);
		var ret = func(arg);
		cache.set(arg, ret);
		return ret;
	};
};
//#endregion
//#region ../../node_modules/@emotion/memoize/dist/emotion-memoize.esm.js
function memoize(fn) {
	var cache = Object.create(null);
	return function(arg) {
		if (cache[arg] === void 0) cache[arg] = fn(arg);
		return cache[arg];
	};
}
//#endregion
//#region ../../node_modules/@emotion/cache/dist/emotion-cache.esm.js
var isBrowser$1 = typeof document !== "undefined";
var identifierWithPointTracking = function identifierWithPointTracking(begin, points, index) {
	var previous = 0;
	var character = 0;
	while (true) {
		previous = character;
		character = peek();
		if (previous === 38 && character === 12) points[index] = 1;
		if (token(character)) break;
		next();
	}
	return slice(begin, position);
};
var toRules = function toRules(parsed, points) {
	var index = -1;
	var character = 44;
	do
		switch (token(character)) {
			case 0:
				if (character === 38 && peek() === 12) points[index] = 1;
				parsed[index] += identifierWithPointTracking(position - 1, points, index);
				break;
			case 2:
				parsed[index] += delimit(character);
				break;
			case 4: if (character === 44) {
				parsed[++index] = peek() === 58 ? "&\f" : "";
				points[index] = parsed[index].length;
				break;
			}
			default: parsed[index] += from(character);
		}
	while (character = next());
	return parsed;
};
var getRules = function getRules(value, points) {
	return dealloc(toRules(alloc(value), points));
};
var fixedElements = /* #__PURE__ */ new WeakMap();
var compat = function compat(element) {
	if (element.type !== "rule" || !element.parent || element.length < 1) return;
	var value = element.value;
	var parent = element.parent;
	var isImplicitRule = element.column === parent.column && element.line === parent.line;
	while (parent.type !== "rule") {
		parent = parent.parent;
		if (!parent) return;
	}
	if (element.props.length === 1 && value.charCodeAt(0) !== 58 && !fixedElements.get(parent)) return;
	if (isImplicitRule) return;
	fixedElements.set(element, true);
	var points = [];
	var rules = getRules(value, points);
	var parentRules = parent.props;
	for (var i = 0, k = 0; i < rules.length; i++) for (var j = 0; j < parentRules.length; j++, k++) element.props[k] = points[i] ? rules[i].replace(/&\f/g, parentRules[j]) : parentRules[j] + " " + rules[i];
};
var removeLabel = function removeLabel(element) {
	if (element.type === "decl") {
		var value = element.value;
		if (value.charCodeAt(0) === 108 && value.charCodeAt(2) === 98) {
			element["return"] = "";
			element.value = "";
		}
	}
};
function prefix(value, length) {
	switch (hash(value, length)) {
		case 5103: return WEBKIT + "print-" + value + value;
		case 5737:
		case 4201:
		case 3177:
		case 3433:
		case 1641:
		case 4457:
		case 2921:
		case 5572:
		case 6356:
		case 5844:
		case 3191:
		case 6645:
		case 3005:
		case 6391:
		case 5879:
		case 5623:
		case 6135:
		case 4599:
		case 4855:
		case 4215:
		case 6389:
		case 5109:
		case 5365:
		case 5621:
		case 3829: return WEBKIT + value + value;
		case 5349:
		case 4246:
		case 4810:
		case 6968:
		case 2756: return WEBKIT + value + MOZ + value + MS + value + value;
		case 6828:
		case 4268: return WEBKIT + value + MS + value + value;
		case 6165: return WEBKIT + value + MS + "flex-" + value + value;
		case 5187: return WEBKIT + value + replace(value, /(\w+).+(:[^]+)/, WEBKIT + "box-$1$2" + MS + "flex-$1$2") + value;
		case 5443: return WEBKIT + value + MS + "flex-item-" + replace(value, /flex-|-self/, "") + value;
		case 4675: return WEBKIT + value + MS + "flex-line-pack" + replace(value, /align-content|flex-|-self/, "") + value;
		case 5548: return WEBKIT + value + MS + replace(value, "shrink", "negative") + value;
		case 5292: return WEBKIT + value + MS + replace(value, "basis", "preferred-size") + value;
		case 6060: return WEBKIT + "box-" + replace(value, "-grow", "") + WEBKIT + value + MS + replace(value, "grow", "positive") + value;
		case 4554: return WEBKIT + replace(value, /([^-])(transform)/g, "$1" + WEBKIT + "$2") + value;
		case 6187: return replace(replace(replace(value, /(zoom-|grab)/, WEBKIT + "$1"), /(image-set)/, WEBKIT + "$1"), value, "") + value;
		case 5495:
		case 3959: return replace(value, /(image-set\([^]*)/, WEBKIT + "$1$`$1");
		case 4968: return replace(replace(value, /(.+:)(flex-)?(.*)/, WEBKIT + "box-pack:$3" + MS + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + WEBKIT + value + value;
		case 4095:
		case 3583:
		case 4068:
		case 2532: return replace(value, /(.+)-inline(.+)/, WEBKIT + "$1$2") + value;
		case 8116:
		case 7059:
		case 5753:
		case 5535:
		case 5445:
		case 5701:
		case 4933:
		case 4677:
		case 5533:
		case 5789:
		case 5021:
		case 4765:
			if (strlen(value) - 1 - length > 6) switch (charat(value, length + 1)) {
				case 109: if (charat(value, length + 4) !== 45) break;
				case 102: return replace(value, /(.+:)(.+)-([^]+)/, "$1" + WEBKIT + "$2-$3$1" + MOZ + (charat(value, length + 3) == 108 ? "$3" : "$2-$3")) + value;
				case 115: return ~indexof(value, "stretch") ? prefix(replace(value, "stretch", "fill-available"), length) + value : value;
			}
			break;
		case 4949: if (charat(value, length + 1) !== 115) break;
		case 6444:
			switch (charat(value, strlen(value) - 3 - (~indexof(value, "!important") && 10))) {
				case 107: return replace(value, ":", ":" + WEBKIT) + value;
				case 101: return replace(value, /(.+:)([^;!]+)(;|!.+)?/, "$1" + WEBKIT + (charat(value, 14) === 45 ? "inline-" : "") + "box$3$1" + WEBKIT + "$2$3$1" + MS + "$2box$3") + value;
			}
			break;
		case 5936:
			switch (charat(value, length + 11)) {
				case 114: return WEBKIT + value + MS + replace(value, /[svh]\w+-[tblr]{2}/, "tb") + value;
				case 108: return WEBKIT + value + MS + replace(value, /[svh]\w+-[tblr]{2}/, "tb-rl") + value;
				case 45: return WEBKIT + value + MS + replace(value, /[svh]\w+-[tblr]{2}/, "lr") + value;
			}
			return WEBKIT + value + MS + value + value;
	}
	return value;
}
var prefixer = function prefixer(element, index, children, callback) {
	if (element.length > -1) {
		if (!element["return"]) switch (element.type) {
			case DECLARATION:
				element["return"] = prefix(element.value, element.length);
				break;
			case KEYFRAMES: return serialize([copy(element, { value: replace(element.value, "@", "@" + WEBKIT) })], callback);
			case RULESET: if (element.length) return combine(element.props, function(value) {
				switch (match(value, /(::plac\w+|:read-\w+)/)) {
					case ":read-only":
					case ":read-write": return serialize([copy(element, { props: [replace(value, /:(read-\w+)/, ":" + MOZ + "$1")] })], callback);
					case "::placeholder": return serialize([
						copy(element, { props: [replace(value, /:(plac\w+)/, ":" + WEBKIT + "input-$1")] }),
						copy(element, { props: [replace(value, /:(plac\w+)/, ":" + MOZ + "$1")] }),
						copy(element, { props: [replace(value, /:(plac\w+)/, MS + "input-$1")] })
					], callback);
				}
				return "";
			});
		}
	}
};
var getServerStylisCache = isBrowser$1 ? void 0 : weakMemoize(function() {
	return memoize(function() {
		return {};
	});
});
var defaultStylisPlugins = [prefixer];
var createCache = function createCache(options) {
	var key = options.key;
	if (isBrowser$1 && key === "css") {
		var ssrStyles = document.querySelectorAll("style[data-emotion]:not([data-s])");
		Array.prototype.forEach.call(ssrStyles, function(node) {
			if (node.getAttribute("data-emotion").indexOf(" ") === -1) return;
			document.head.appendChild(node);
			node.setAttribute("data-s", "");
		});
	}
	var stylisPlugins = options.stylisPlugins || defaultStylisPlugins;
	var inserted = {};
	var container;
	var nodesToHydrate = [];
	if (isBrowser$1) {
		container = options.container || document.head;
		Array.prototype.forEach.call(document.querySelectorAll("style[data-emotion^=\"" + key + " \"]"), function(node) {
			var attrib = node.getAttribute("data-emotion").split(" ");
			for (var i = 1; i < attrib.length; i++) inserted[attrib[i]] = true;
			nodesToHydrate.push(node);
		});
	}
	var _insert;
	var omnipresentPlugins = [compat, removeLabel];
	if (!getServerStylisCache) {
		var currentSheet;
		var finalizingPlugins = [stringify, rulesheet(function(rule) {
			currentSheet.insert(rule);
		})];
		var serializer = middleware(omnipresentPlugins.concat(stylisPlugins, finalizingPlugins));
		var stylis = function stylis(styles) {
			return serialize(compile(styles), serializer);
		};
		_insert = function insert(selector, serialized, sheet, shouldCache) {
			currentSheet = sheet;
			stylis(selector ? selector + "{" + serialized.styles + "}" : serialized.styles);
			if (shouldCache) cache.inserted[serialized.name] = true;
		};
	} else {
		var _finalizingPlugins = [stringify];
		var _serializer = middleware(omnipresentPlugins.concat(stylisPlugins, _finalizingPlugins));
		var _stylis = function _stylis(styles) {
			return serialize(compile(styles), _serializer);
		};
		var serverStylisCache = getServerStylisCache(stylisPlugins)(key);
		var getRules = function getRules(selector, serialized) {
			var name = serialized.name;
			if (serverStylisCache[name] === void 0) serverStylisCache[name] = _stylis(selector ? selector + "{" + serialized.styles + "}" : serialized.styles);
			return serverStylisCache[name];
		};
		_insert = function _insert(selector, serialized, sheet, shouldCache) {
			var name = serialized.name;
			var rules = getRules(selector, serialized);
			if (cache.compat === void 0) {
				if (shouldCache) cache.inserted[name] = true;
				return rules;
			} else if (shouldCache) cache.inserted[name] = rules;
			else return rules;
		};
	}
	var cache = {
		key,
		sheet: new StyleSheet$1({
			key,
			container,
			nonce: options.nonce,
			speedy: options.speedy,
			prepend: options.prepend,
			insertionPoint: options.insertionPoint
		}),
		nonce: options.nonce,
		inserted,
		registered: {},
		insert: _insert
	};
	cache.sheet.hydrate(nodesToHydrate);
	return cache;
};
//#endregion
//#region ../../node_modules/@babel/runtime/helpers/esm/extends.js
function _extends() {
	return _extends = Object.assign ? Object.assign.bind() : function(n) {
		for (var e = 1; e < arguments.length; e++) {
			var t = arguments[e];
			for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]);
		}
		return n;
	}, _extends.apply(null, arguments);
}
//#endregion
//#region ../../node_modules/react-is/cjs/react-is.production.min.js
/** @license React v16.13.1
* react-is.production.min.js
*
* Copyright (c) Facebook, Inc. and its affiliates.
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/
var require_react_is_production_min = /* @__PURE__ */ __commonJSMin(((exports) => {
	var b = "function" === typeof Symbol && Symbol.for, c = b ? Symbol.for("react.element") : 60103, d = b ? Symbol.for("react.portal") : 60106, e = b ? Symbol.for("react.fragment") : 60107, f = b ? Symbol.for("react.strict_mode") : 60108, g = b ? Symbol.for("react.profiler") : 60114, h = b ? Symbol.for("react.provider") : 60109, k = b ? Symbol.for("react.context") : 60110, l = b ? Symbol.for("react.async_mode") : 60111, m = b ? Symbol.for("react.concurrent_mode") : 60111, n = b ? Symbol.for("react.forward_ref") : 60112, p = b ? Symbol.for("react.suspense") : 60113, q = b ? Symbol.for("react.suspense_list") : 60120, r = b ? Symbol.for("react.memo") : 60115, t = b ? Symbol.for("react.lazy") : 60116, v = b ? Symbol.for("react.block") : 60121, w = b ? Symbol.for("react.fundamental") : 60117, x = b ? Symbol.for("react.responder") : 60118, y = b ? Symbol.for("react.scope") : 60119;
	function z(a) {
		if ("object" === typeof a && null !== a) {
			var u = a.$$typeof;
			switch (u) {
				case c: switch (a = a.type, a) {
					case l:
					case m:
					case e:
					case g:
					case f:
					case p: return a;
					default: switch (a = a && a.$$typeof, a) {
						case k:
						case n:
						case t:
						case r:
						case h: return a;
						default: return u;
					}
				}
				case d: return u;
			}
		}
	}
	function A(a) {
		return z(a) === m;
	}
	exports.AsyncMode = l;
	exports.ConcurrentMode = m;
	exports.ContextConsumer = k;
	exports.ContextProvider = h;
	exports.Element = c;
	exports.ForwardRef = n;
	exports.Fragment = e;
	exports.Lazy = t;
	exports.Memo = r;
	exports.Portal = d;
	exports.Profiler = g;
	exports.StrictMode = f;
	exports.Suspense = p;
	exports.isAsyncMode = function(a) {
		return A(a) || z(a) === l;
	};
	exports.isConcurrentMode = A;
	exports.isContextConsumer = function(a) {
		return z(a) === k;
	};
	exports.isContextProvider = function(a) {
		return z(a) === h;
	};
	exports.isElement = function(a) {
		return "object" === typeof a && null !== a && a.$$typeof === c;
	};
	exports.isForwardRef = function(a) {
		return z(a) === n;
	};
	exports.isFragment = function(a) {
		return z(a) === e;
	};
	exports.isLazy = function(a) {
		return z(a) === t;
	};
	exports.isMemo = function(a) {
		return z(a) === r;
	};
	exports.isPortal = function(a) {
		return z(a) === d;
	};
	exports.isProfiler = function(a) {
		return z(a) === g;
	};
	exports.isStrictMode = function(a) {
		return z(a) === f;
	};
	exports.isSuspense = function(a) {
		return z(a) === p;
	};
	exports.isValidElementType = function(a) {
		return "string" === typeof a || "function" === typeof a || a === e || a === m || a === g || a === f || a === p || a === q || "object" === typeof a && null !== a && (a.$$typeof === t || a.$$typeof === r || a.$$typeof === h || a.$$typeof === k || a.$$typeof === n || a.$$typeof === w || a.$$typeof === x || a.$$typeof === y || a.$$typeof === v);
	};
	exports.typeOf = z;
}));
//#endregion
//#region ../../node_modules/react-is/cjs/react-is.development.js
/** @license React v16.13.1
* react-is.development.js
*
* Copyright (c) Facebook, Inc. and its affiliates.
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/
var require_react_is_development = /* @__PURE__ */ __commonJSMin(((exports) => {
	if (process.env.NODE_ENV !== "production") (function() {
		"use strict";
		var hasSymbol = typeof Symbol === "function" && Symbol.for;
		var REACT_ELEMENT_TYPE = hasSymbol ? Symbol.for("react.element") : 60103;
		var REACT_PORTAL_TYPE = hasSymbol ? Symbol.for("react.portal") : 60106;
		var REACT_FRAGMENT_TYPE = hasSymbol ? Symbol.for("react.fragment") : 60107;
		var REACT_STRICT_MODE_TYPE = hasSymbol ? Symbol.for("react.strict_mode") : 60108;
		var REACT_PROFILER_TYPE = hasSymbol ? Symbol.for("react.profiler") : 60114;
		var REACT_PROVIDER_TYPE = hasSymbol ? Symbol.for("react.provider") : 60109;
		var REACT_CONTEXT_TYPE = hasSymbol ? Symbol.for("react.context") : 60110;
		var REACT_ASYNC_MODE_TYPE = hasSymbol ? Symbol.for("react.async_mode") : 60111;
		var REACT_CONCURRENT_MODE_TYPE = hasSymbol ? Symbol.for("react.concurrent_mode") : 60111;
		var REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for("react.forward_ref") : 60112;
		var REACT_SUSPENSE_TYPE = hasSymbol ? Symbol.for("react.suspense") : 60113;
		var REACT_SUSPENSE_LIST_TYPE = hasSymbol ? Symbol.for("react.suspense_list") : 60120;
		var REACT_MEMO_TYPE = hasSymbol ? Symbol.for("react.memo") : 60115;
		var REACT_LAZY_TYPE = hasSymbol ? Symbol.for("react.lazy") : 60116;
		var REACT_BLOCK_TYPE = hasSymbol ? Symbol.for("react.block") : 60121;
		var REACT_FUNDAMENTAL_TYPE = hasSymbol ? Symbol.for("react.fundamental") : 60117;
		var REACT_RESPONDER_TYPE = hasSymbol ? Symbol.for("react.responder") : 60118;
		var REACT_SCOPE_TYPE = hasSymbol ? Symbol.for("react.scope") : 60119;
		function isValidElementType(type) {
			return typeof type === "string" || typeof type === "function" || type === REACT_FRAGMENT_TYPE || type === REACT_CONCURRENT_MODE_TYPE || type === REACT_PROFILER_TYPE || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || typeof type === "object" && type !== null && (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || type.$$typeof === REACT_FUNDAMENTAL_TYPE || type.$$typeof === REACT_RESPONDER_TYPE || type.$$typeof === REACT_SCOPE_TYPE || type.$$typeof === REACT_BLOCK_TYPE);
		}
		function typeOf(object) {
			if (typeof object === "object" && object !== null) {
				var $$typeof = object.$$typeof;
				switch ($$typeof) {
					case REACT_ELEMENT_TYPE:
						var type = object.type;
						switch (type) {
							case REACT_ASYNC_MODE_TYPE:
							case REACT_CONCURRENT_MODE_TYPE:
							case REACT_FRAGMENT_TYPE:
							case REACT_PROFILER_TYPE:
							case REACT_STRICT_MODE_TYPE:
							case REACT_SUSPENSE_TYPE: return type;
							default:
								var $$typeofType = type && type.$$typeof;
								switch ($$typeofType) {
									case REACT_CONTEXT_TYPE:
									case REACT_FORWARD_REF_TYPE:
									case REACT_LAZY_TYPE:
									case REACT_MEMO_TYPE:
									case REACT_PROVIDER_TYPE: return $$typeofType;
									default: return $$typeof;
								}
						}
					case REACT_PORTAL_TYPE: return $$typeof;
				}
			}
		}
		var AsyncMode = REACT_ASYNC_MODE_TYPE;
		var ConcurrentMode = REACT_CONCURRENT_MODE_TYPE;
		var ContextConsumer = REACT_CONTEXT_TYPE;
		var ContextProvider = REACT_PROVIDER_TYPE;
		var Element = REACT_ELEMENT_TYPE;
		var ForwardRef = REACT_FORWARD_REF_TYPE;
		var Fragment = REACT_FRAGMENT_TYPE;
		var Lazy = REACT_LAZY_TYPE;
		var Memo = REACT_MEMO_TYPE;
		var Portal = REACT_PORTAL_TYPE;
		var Profiler = REACT_PROFILER_TYPE;
		var StrictMode = REACT_STRICT_MODE_TYPE;
		var Suspense = REACT_SUSPENSE_TYPE;
		var hasWarnedAboutDeprecatedIsAsyncMode = false;
		function isAsyncMode(object) {
			if (!hasWarnedAboutDeprecatedIsAsyncMode) {
				hasWarnedAboutDeprecatedIsAsyncMode = true;
				console["warn"]("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.");
			}
			return isConcurrentMode(object) || typeOf(object) === REACT_ASYNC_MODE_TYPE;
		}
		function isConcurrentMode(object) {
			return typeOf(object) === REACT_CONCURRENT_MODE_TYPE;
		}
		function isContextConsumer(object) {
			return typeOf(object) === REACT_CONTEXT_TYPE;
		}
		function isContextProvider(object) {
			return typeOf(object) === REACT_PROVIDER_TYPE;
		}
		function isElement(object) {
			return typeof object === "object" && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
		}
		function isForwardRef(object) {
			return typeOf(object) === REACT_FORWARD_REF_TYPE;
		}
		function isFragment(object) {
			return typeOf(object) === REACT_FRAGMENT_TYPE;
		}
		function isLazy(object) {
			return typeOf(object) === REACT_LAZY_TYPE;
		}
		function isMemo(object) {
			return typeOf(object) === REACT_MEMO_TYPE;
		}
		function isPortal(object) {
			return typeOf(object) === REACT_PORTAL_TYPE;
		}
		function isProfiler(object) {
			return typeOf(object) === REACT_PROFILER_TYPE;
		}
		function isStrictMode(object) {
			return typeOf(object) === REACT_STRICT_MODE_TYPE;
		}
		function isSuspense(object) {
			return typeOf(object) === REACT_SUSPENSE_TYPE;
		}
		exports.AsyncMode = AsyncMode;
		exports.ConcurrentMode = ConcurrentMode;
		exports.ContextConsumer = ContextConsumer;
		exports.ContextProvider = ContextProvider;
		exports.Element = Element;
		exports.ForwardRef = ForwardRef;
		exports.Fragment = Fragment;
		exports.Lazy = Lazy;
		exports.Memo = Memo;
		exports.Portal = Portal;
		exports.Profiler = Profiler;
		exports.StrictMode = StrictMode;
		exports.Suspense = Suspense;
		exports.isAsyncMode = isAsyncMode;
		exports.isConcurrentMode = isConcurrentMode;
		exports.isContextConsumer = isContextConsumer;
		exports.isContextProvider = isContextProvider;
		exports.isElement = isElement;
		exports.isForwardRef = isForwardRef;
		exports.isFragment = isFragment;
		exports.isLazy = isLazy;
		exports.isMemo = isMemo;
		exports.isPortal = isPortal;
		exports.isProfiler = isProfiler;
		exports.isStrictMode = isStrictMode;
		exports.isSuspense = isSuspense;
		exports.isValidElementType = isValidElementType;
		exports.typeOf = typeOf;
	})();
}));
//#endregion
//#region ../../node_modules/react-is/index.js
var require_react_is = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	if (process.env.NODE_ENV === "production") module.exports = require_react_is_production_min();
	else module.exports = require_react_is_development();
}));
//#endregion
//#region ../../node_modules/@emotion/react/_isolated-hnrs/dist/emotion-react-_isolated-hnrs.esm.js
var import_hoist_non_react_statics_cjs = /* @__PURE__ */ __toESM((/* @__PURE__ */ __commonJSMin(((exports, module) => {
	var reactIs = require_react_is();
	/**
	* Copyright 2015, Yahoo! Inc.
	* Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
	*/
	var REACT_STATICS = {
		childContextTypes: true,
		contextType: true,
		contextTypes: true,
		defaultProps: true,
		displayName: true,
		getDefaultProps: true,
		getDerivedStateFromError: true,
		getDerivedStateFromProps: true,
		mixins: true,
		propTypes: true,
		type: true
	};
	var KNOWN_STATICS = {
		name: true,
		length: true,
		prototype: true,
		caller: true,
		callee: true,
		arguments: true,
		arity: true
	};
	var FORWARD_REF_STATICS = {
		"$$typeof": true,
		render: true,
		defaultProps: true,
		displayName: true,
		propTypes: true
	};
	var MEMO_STATICS = {
		"$$typeof": true,
		compare: true,
		defaultProps: true,
		displayName: true,
		propTypes: true,
		type: true
	};
	var TYPE_STATICS = {};
	TYPE_STATICS[reactIs.ForwardRef] = FORWARD_REF_STATICS;
	TYPE_STATICS[reactIs.Memo] = MEMO_STATICS;
	function getStatics(component) {
		if (reactIs.isMemo(component)) return MEMO_STATICS;
		return TYPE_STATICS[component["$$typeof"]] || REACT_STATICS;
	}
	var defineProperty = Object.defineProperty;
	var getOwnPropertyNames = Object.getOwnPropertyNames;
	var getOwnPropertySymbols = Object.getOwnPropertySymbols;
	var getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor;
	var getPrototypeOf = Object.getPrototypeOf;
	var objectPrototype = Object.prototype;
	function hoistNonReactStatics(targetComponent, sourceComponent, blacklist) {
		if (typeof sourceComponent !== "string") {
			if (objectPrototype) {
				var inheritedComponent = getPrototypeOf(sourceComponent);
				if (inheritedComponent && inheritedComponent !== objectPrototype) hoistNonReactStatics(targetComponent, inheritedComponent, blacklist);
			}
			var keys = getOwnPropertyNames(sourceComponent);
			if (getOwnPropertySymbols) keys = keys.concat(getOwnPropertySymbols(sourceComponent));
			var targetStatics = getStatics(targetComponent);
			var sourceStatics = getStatics(sourceComponent);
			for (var i = 0; i < keys.length; ++i) {
				var key = keys[i];
				if (!KNOWN_STATICS[key] && !(blacklist && blacklist[key]) && !(sourceStatics && sourceStatics[key]) && !(targetStatics && targetStatics[key])) {
					var descriptor = getOwnPropertyDescriptor(sourceComponent, key);
					try {
						defineProperty(targetComponent, key, descriptor);
					} catch (e) {}
				}
			}
		}
		return targetComponent;
	}
	module.exports = hoistNonReactStatics;
})))());
var hoistNonReactStatics = (function(targetComponent, sourceComponent) {
	return (0, import_hoist_non_react_statics_cjs.default)(targetComponent, sourceComponent);
});
react["useInsertionEffect"] && react["useInsertionEffect"] || react.useLayoutEffect;
//#endregion
//#region ../../node_modules/@emotion/react/dist/emotion-element-d59e098f.esm.js
var isBrowser = typeof document !== "undefined";
(/* @__PURE__ */ react.createContext(typeof HTMLElement !== "undefined" ? /* #__PURE__ */ createCache({ key: "css" }) : null)).Provider;
if (!isBrowser);
var ThemeContext = /* #__PURE__ */ react.createContext({});
var useTheme = function useTheme() {
	return react.useContext(ThemeContext);
};
var getTheme = function getTheme(outerTheme, theme) {
	if (typeof theme === "function") return theme(outerTheme);
	return _extends({}, outerTheme, theme);
};
var createCacheWithTheme = /* #__PURE__ */ weakMemoize(function(outerTheme) {
	return weakMemoize(function(theme) {
		return getTheme(outerTheme, theme);
	});
});
var ThemeProvider = function ThemeProvider(props) {
	var theme = react.useContext(ThemeContext);
	if (props.theme !== theme) theme = createCacheWithTheme(theme)(props.theme);
	return /*#__PURE__*/ react.createElement(ThemeContext.Provider, { value: theme }, props.children);
};
function withTheme(Component) {
	var componentName = Component.displayName || Component.name || "Component";
	var WithTheme = /*#__PURE__*/ react.forwardRef(function render(props, ref) {
		var theme = react.useContext(ThemeContext);
		return /*#__PURE__*/ react.createElement(Component, _extends({
			theme,
			ref
		}, props));
	});
	WithTheme.displayName = "WithTheme(" + componentName + ")";
	return hoistNonReactStatics(WithTheme, Component);
}
({}).hasOwnProperty;
//#endregion
//#region ../../node_modules/@emotion/primitives-core/dist/emotion-primitives-core.esm.js
var import_css_to_react_native = /* @__PURE__ */ __toESM(require_css_to_react_native());
function interleave(vals) {
	var strings = vals[0];
	var finalArray = [strings[0]];
	for (var i = 1, len = vals.length; i < len; i++) {
		finalArray.push(vals[i]);
		if (strings[i] !== void 0) finalArray.push(strings[i]);
	}
	return finalArray;
}
var styles;
var generated = {};
var buffer = "";
var lastType;
function handleInterpolation(interpolation, i, arr) {
	var type = typeof interpolation;
	if (type === "string") interpolation = interpolation.replace(/\/\*[\s\S]*?\*\/|\/\/.*$/gm, "");
	if (type === "function") {
		if (this === void 0);
		else handleInterpolation.call(this, interpolation(this), i, arr);
		return;
	}
	var isIrrelevant = interpolation == null || type === "boolean";
	var isRnStyle = type === "object" && !Array.isArray(interpolation) || type === "number";
	if (lastType === "string" && (isRnStyle || isIrrelevant)) {
		var converted = convertStyles(buffer);
		if (converted !== void 0) styles.push(converted);
		buffer = "";
	}
	if (isIrrelevant) return;
	if (type === "string") {
		buffer += interpolation;
		if (arr.length - 1 === i) {
			var _converted = convertStyles(buffer);
			if (_converted !== void 0) styles.push(_converted);
			buffer = "";
		}
	}
	if (isRnStyle) styles.push(interpolation);
	if (Array.isArray(interpolation)) interpolation.forEach(handleInterpolation, this);
	lastType = type;
}
function createCss(StyleSheet) {
	return function css() {
		var prevBuffer = buffer;
		var vals;
		styles = [];
		buffer = "";
		lastType = void 0;
		for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) args[_key] = arguments[_key];
		if (args[0] == null || args[0].raw === void 0) vals = args;
		else vals = interleave(args);
		try {
			vals.forEach(handleInterpolation, this);
		} finally {
			buffer = prevBuffer;
		}
		var hash = JSON.stringify(styles);
		if (!generated[hash]) generated[hash] = StyleSheet.create({ generated: StyleSheet.flatten(styles) }).generated;
		return generated[hash];
	};
}
var propertyValuePattern = /\s*([^\s]+)\s*:\s*(.+?)\s*$/;
function convertPropertyValue(style) {
	var match = propertyValuePattern.exec(style);
	if (match !== null && match[2] !== " ") {
		match.shift();
		this.push(match);
	}
}
function convertStyles(str) {
	if (str.trim() === "") return;
	var stylePairs = [];
	str.split(";").forEach(convertPropertyValue, stylePairs);
	try {
		return (0, import_css_to_react_native.default)(stylePairs);
	} catch (error) {
		var msg = error == null ? void 0 : error.message;
		if (msg && msg.includes("Failed to parse declaration")) {
			var values = msg.replace("Failed to parse declaration ", "").replace(/"/g, "").trim().split(":");
			var errorMsg = "'" + values[0] + "' shorthand property requires units for example - " + values[0] + ": 20px or " + values[0] + ": 10px 20px 40px 50px";
			console.error(errorMsg);
		}
	}
}
var testOmitPropsOnComponent = function testOmitPropsOnComponent(prop) {
	return prop !== "theme" && prop !== "as";
};
function createStyled(StyleSheet, options) {
	var _options$getShouldFor;
	var getShouldForwardProp = (_options$getShouldFor = options == null ? void 0 : options.getShouldForwardProp) != null ? _options$getShouldFor : function() {
		return testOmitPropsOnComponent;
	};
	var css = createCss(StyleSheet);
	return function createEmotion(component, options) {
		var shouldForwardProp = options && options.shouldForwardProp ? options.shouldForwardProp : void 0;
		var defaultShouldForwardProp = shouldForwardProp || getShouldForwardProp(component);
		var shouldUseAs = !defaultShouldForwardProp("as");
		return function createStyledComponent() {
			var styles;
			for (var _len = arguments.length, rawStyles = new Array(_len), _key = 0; _key < _len; _key++) rawStyles[_key] = arguments[_key];
			if (rawStyles[0] == null || rawStyles[0].raw === void 0) styles = rawStyles;
			else styles = interleave(rawStyles);
			var Styled = /*#__PURE__*/ react.forwardRef(function(props, ref) {
				var finalTag = shouldUseAs && props.as || component;
				var mergedProps = props;
				if (props.theme == null) {
					mergedProps = {};
					for (var key in props) mergedProps[key] = props[key];
					mergedProps.theme = react.useContext(ThemeContext);
				}
				var finalShouldForwardProp = shouldUseAs && shouldForwardProp === void 0 ? getShouldForwardProp(finalTag) : defaultShouldForwardProp;
				var newProps = {};
				for (var _key2 in props) {
					if (shouldUseAs && _key2 === "as") continue;
					if (finalShouldForwardProp(_key2)) newProps[_key2] = props[_key2];
				}
				newProps.style = [css.apply(mergedProps, styles), props.style];
				if (ref) newProps.ref = ref;
				return /*#__PURE__*/ react.createElement(finalTag, newProps);
			});
			Styled.displayName = "emotion(" + getDisplayName(component) + ")";
			var withComponent = function withComponent(newComponent) {
				return createEmotion(newComponent).apply(void 0, styles);
			};
			var castedStyled = Styled;
			castedStyled.withComponent = withComponent;
			return castedStyled;
		};
	};
}
var getDisplayName = function getDisplayName(primitive) {
	return typeof primitive === "string" ? primitive : primitive.displayName || primitive.name || "Styled";
};
//#endregion
//#region ../../node_modules/@emotion/native/dist/emotion-native.esm.js
/**
* a function that returns a styled component which render styles in React Native
*/
var styled = createStyled(react_native.StyleSheet);
react_native.StyleSheet;
var index = [
	"ActivityIndicator",
	"Button",
	"DatePickerIOS",
	"DrawerLayoutAndroid",
	"FlatList",
	"Image",
	"ImageBackground",
	"KeyboardAvoidingView",
	"ListView",
	"Modal",
	"NavigatorIOS",
	"Picker",
	"PickerIOS",
	"Pressable",
	"ProgressBarAndroid",
	"ProgressViewIOS",
	"RecyclerViewBackedScrollView",
	"RefreshControl",
	"SafeAreaView",
	"ScrollView",
	"SectionList",
	"SegmentedControlIOS",
	"Slider",
	"SnapshotViewIOS",
	"StatusBar",
	"SwipeableListView",
	"Switch",
	"SwitchIOS",
	"TabBarIOS",
	"Text",
	"TextInput",
	"ToolbarAndroid",
	"TouchableHighlight",
	"TouchableNativeFeedback",
	"TouchableOpacity",
	"TouchableWithoutFeedback",
	"View",
	"ViewPagerAndroid"
].reduce(function(acc, comp) {
	return Object.defineProperty(acc, comp, {
		enumerable: true,
		configurable: false,
		get: function get() {
			return styled(react_native[comp]);
		}
	});
}, styled);
//#endregion
//#region src/web-theme.ts
const color = {
	primary: "#FF4785",
	secondary: "#029CFD",
	secondaryLighter: "rgba(2, 157, 253, 0.9)",
	tertiary: "#FAFBFC",
	ancillary: "#22a699",
	orange: "#FC521F",
	gold: "#FFAE00",
	green: "#66BF3C",
	seafoam: "#37D5D3",
	purple: "#6F2CAC",
	ultraviolet: "#2A0481",
	lightest: "#FFFFFF",
	lighter: "#F7FAFC",
	light: "#EEF3F6",
	mediumlight: "#ECF4F9",
	medium: "#D9E8F2",
	mediumdark: "#73828C",
	dark: "#5C6870",
	darker: "#454E54",
	darkest: "#2E3438",
	border: "hsla(203, 50%, 30%, 0.15)",
	positive: "#66BF3C",
	negative: "#FF4400",
	warning: "#E69D00",
	critical: "#FFFFFF",
	defaultText: "#2E3438",
	inverseText: "#FFFFFF",
	positiveText: "#448028",
	negativeText: "#D43900",
	warningText: "#A15C20"
};
const background = {
	app: "#F6F9FC",
	bar: color.lightest,
	content: color.lightest,
	preview: color.lightest,
	gridCellSize: 10,
	hoverable: (0, polished.transparentize)(.9, color.secondary),
	positive: "#E1FFD4",
	negative: "#FEDED2",
	warning: "#FFF5CF",
	critical: "#FF4400"
};
const light = {
	base: "light",
	colorPrimary: "#FF4785",
	colorSecondary: "#029CFD",
	appBg: background.app,
	appContentBg: color.lightest,
	appPreviewBg: color.lightest,
	appBorderColor: color.border,
	appBorderRadius: 4,
	textColor: color.darkest,
	textInverseColor: color.lightest,
	textMutedColor: color.dark,
	barTextColor: color.mediumdark,
	barHoverColor: color.secondary,
	barSelectedColor: color.secondary,
	barBg: color.lightest,
	buttonBg: background.app,
	buttonBorder: color.medium,
	booleanBg: color.mediumlight,
	booleanSelectedBg: color.lightest,
	inputBg: color.lightest,
	inputBorder: color.border,
	inputTextColor: color.darkest,
	inputBorderRadius: 4
};
const dark = {
	base: "dark",
	colorPrimary: "#FF4785",
	colorSecondary: "#029CFD",
	appBg: "#222425",
	appContentBg: "#1B1C1D",
	appPreviewBg: color.lightest,
	appBorderColor: "rgba(255,255,255,.1)",
	appBorderRadius: 4,
	textColor: "#C9CDCF",
	textInverseColor: "#222425",
	textMutedColor: "#798186",
	barTextColor: color.mediumdark,
	barHoverColor: color.secondary,
	barSelectedColor: color.secondary,
	barBg: "#292C2E",
	buttonBg: "#222425",
	buttonBorder: "rgba(255,255,255,.1)",
	booleanBg: "#222425",
	booleanSelectedBg: "#2E3438",
	inputBg: "#1B1C1D",
	inputBorder: "rgba(255,255,255,.1)",
	inputTextColor: color.lightest,
	inputBorderRadius: 4
};
//#endregion
//#region src/theme.ts
const typography = {
	weight: {
		regular: "400",
		bold: "700"
	},
	size: {
		s1: 12,
		s2: 14,
		s3: 16,
		m1: 20,
		m2: 24,
		m3: 28,
		l1: 32,
		l2: 40,
		l3: 48,
		code: 90
	}
};
const theme = {
	base: "light",
	textMutedColor: color.dark,
	color: {
		primary: light.colorPrimary,
		secondary: light.colorSecondary,
		secondaryLighter: color.secondaryLighter,
		tertiary: color.tertiary,
		ancillary: color.ancillary,
		orange: color.orange,
		gold: color.gold,
		green: color.green,
		seafoam: color.seafoam,
		purple: color.purple,
		ultraviolet: color.ultraviolet,
		lightest: color.lightest,
		lighter: color.lighter,
		light: color.light,
		mediumlight: color.mediumlight,
		medium: color.medium,
		mediumdark: color.mediumdark,
		dark: color.dark,
		darker: color.darker,
		darkest: color.darkest,
		border: color.border,
		positive: color.positive,
		negative: color.negative,
		warning: color.warning,
		critical: color.critical,
		defaultText: light.textColor || color.darkest,
		inverseText: light.textInverseColor || color.lightest,
		positiveText: color.positiveText,
		negativeText: color.negativeText,
		warningText: color.warningText
	},
	background: {
		app: light.appBg,
		bar: light.barBg,
		content: light.appContentBg,
		preview: light.appPreviewBg,
		gridCellSize: light.gridCellSize || background.gridCellSize,
		hoverable: background.hoverable,
		positive: background.positive,
		negative: background.negative,
		warning: background.warning,
		critical: background.critical
	},
	typography: {
		weight: typography.weight,
		size: typography.size
	},
	input: {
		background: light.inputBg,
		border: light.inputBorder,
		borderRadius: light.inputBorderRadius,
		color: light.inputTextColor,
		paddingHorizontal: 10,
		paddingVertical: 6
	},
	button: {
		background: light.buttonBg || light.inputBg,
		border: light.buttonBorder || light.inputBorder
	},
	boolean: {
		background: light.booleanBg || light.inputBorder,
		selectedBackground: light.booleanSelectedBg || light.inputBg
	},
	layoutMargin: 10,
	appBorderColor: light.appBorderColor,
	appBorderRadius: light.appBorderRadius,
	barTextColor: light.barTextColor,
	barHoverColor: light.barHoverColor || light.colorSecondary,
	barSelectedColor: light.barSelectedColor || light.colorSecondary,
	barBg: light.barBg,
	brand: void 0
};
const darkTheme = {
	base: "dark",
	textMutedColor: "#798186",
	color: {
		primary: dark.colorPrimary,
		secondary: dark.colorSecondary,
		secondaryLighter: color.secondaryLighter,
		tertiary: color.tertiary,
		ancillary: color.ancillary,
		orange: color.orange,
		gold: color.gold,
		green: color.green,
		seafoam: color.seafoam,
		purple: color.purple,
		ultraviolet: color.ultraviolet,
		lightest: color.lightest,
		lighter: color.lighter,
		light: color.light,
		mediumlight: color.mediumlight,
		medium: color.medium,
		mediumdark: color.mediumdark,
		dark: color.dark,
		darker: color.darker,
		darkest: color.darkest,
		border: color.border,
		positive: color.positive,
		negative: color.negative,
		warning: color.warning,
		critical: color.critical,
		defaultText: dark.textColor || color.darkest,
		inverseText: dark.textInverseColor || color.lightest,
		positiveText: color.positiveText,
		negativeText: color.negativeText,
		warningText: color.warningText
	},
	background: {
		app: dark.appBg,
		bar: dark.barBg,
		content: dark.appContentBg,
		preview: dark.appPreviewBg,
		gridCellSize: dark.gridCellSize || background.gridCellSize,
		hoverable: background.hoverable,
		positive: background.positive,
		negative: background.negative,
		warning: background.warning,
		critical: background.critical
	},
	typography: {
		weight: typography.weight,
		size: typography.size
	},
	input: {
		background: dark.inputBg,
		border: dark.inputBorder,
		borderRadius: dark.inputBorderRadius,
		color: dark.inputTextColor,
		paddingHorizontal: 10,
		paddingVertical: 6
	},
	button: {
		background: dark.buttonBg || dark.inputBg,
		border: dark.buttonBorder || dark.inputBorder
	},
	boolean: {
		background: dark.booleanBg || dark.inputBorder,
		selectedBackground: dark.booleanSelectedBg || dark.inputBg
	},
	layoutMargin: 10,
	appBorderColor: dark.appBorderColor,
	appBorderRadius: dark.appBorderRadius,
	barTextColor: dark.barTextColor,
	barHoverColor: dark.barHoverColor || dark.colorSecondary,
	barSelectedColor: dark.barSelectedColor || dark.colorSecondary,
	barBg: dark.barBg,
	brand: void 0
};
//#endregion
exports.ThemeProvider = ThemeProvider;
exports.darkTheme = darkTheme;
exports.styled = index;
exports.theme = theme;
exports.useTheme = useTheme;
exports.withTheme = withTheme;
