// @storybook/addon-ondevice-actions@10.4.7/dist/components/ActionLogger/Inspect.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import React, { useState } from 'react';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import { styled } from '@storybook/react-native-theming';
const DefaultText = styled.Text(({ theme }) => ({
    color: theme.color.defaultText,
}));
const ObjectNameText = styled.Text(({ theme }) => ({
    color: theme.color.secondary,
}));
const MutedText = styled.Text(({ theme }) => ({
    color: theme.textMutedColor,
}));
const StringText = styled.Text(({ theme }) => ({
    color: theme.color.orange,
}));
const NumberText = styled.Text(({ theme }) => ({
    color: theme.color.green,
}));
const BooleanText = styled.Text(({ theme }) => ({
    color: theme.color.seafoam,
}));
const FunctionText = styled.Text(({ theme }) => ({
    color: theme.color.purple,
}));
const ArrowText = styled.Text(({ theme, visible }) => ({
    color: visible ? theme.textMutedColor : 'transparent',
    paddingRight: 8,
}));
const Inspect = ({ name, value }) => {
    const [expanded, setExpanded] = useState(false);
    const canExpand = name &&
        ((Array.isArray(value) && value.length) ||
            (value && typeof value === 'object' && Object.keys(value).length));
    const toggleExpanded = React.useCallback(() => {
        if (canExpand) {
            setExpanded((currentValue) => !currentValue);
        }
    }, [canExpand]);
    const toggle = _jsx(ArrowText, { visible: !!canExpand, children: expanded ? '▼' : '▶' });
    const nameComponent = name ? _jsx(ObjectNameText, { children: name }) : null;
    if (Array.isArray(value)) {
        if (name) {
            return (_jsxs(_Fragment, { children: [_jsxs(TouchableOpacity, { onPress: toggleExpanded, style: styles.row, children: [toggle, nameComponent, _jsx(DefaultText, { children: `: ${value.length === 0 ? '[]' : expanded ? '[' : '[...]'}` })] }), expanded ? (_jsxs(View, { style: styles.expanded, children: [value.map((v, i) => (_jsx(View, { style: styles.expanded, children: _jsx(Inspect, { value: v }) }, i))), _jsx(View, { style: styles.spacingLeft, children: _jsx(DefaultText, { children: "]" }) })] })) : null] }));
        }
        return (_jsxs(_Fragment, { children: [_jsx(DefaultText, { children: "[" }), value.map((v, i) => (_jsx(View, { style: styles.spacingLeft, children: _jsx(Inspect, { value: v }) }, i))), _jsx(DefaultText, { children: "]" })] }));
    }
    if (value && typeof value === 'object' && !(value instanceof RegExp)) {
        if (name) {
            return (_jsxs(_Fragment, { children: [_jsxs(TouchableOpacity, { style: styles.row, onPress: toggleExpanded, children: [toggle, nameComponent, _jsx(DefaultText, { children: `: ${Object.keys(value).length === 0 ? '{}' : expanded ? '{' : '{...}'}` })] }), expanded ? (_jsxs(View, { style: styles.expanded, children: [Object.entries(value).map(([key, v]) => (_jsx(View, { children: _jsx(Inspect, { name: key, value: v }) }, key))), _jsx(View, { style: styles.spacingLeft, children: _jsx(DefaultText, { children: '}' }) })] })) : null] }));
        }
        return (_jsxs(_Fragment, { children: [_jsx(DefaultText, { children: '{' }), Object.entries(value).map(([key, v]) => (_jsx(View, { children: _jsx(Inspect, { name: key, value: v }) }, key))), _jsx(DefaultText, { children: '}' })] }));
    }
    if (name) {
        return (_jsxs(View, { style: styles.row, children: [toggle, nameComponent, _jsx(DefaultText, { children: ": " }), _jsx(Value, { value: value })] }));
    }
    return _jsx(Value, { value: value });
};
function Value({ value }) {
    if (value === null) {
        return _jsx(MutedText, { children: "null" });
    }
    if (value === undefined) {
        return _jsx(MutedText, { children: "undefined" });
    }
    if (value instanceof RegExp) {
        return _jsx(StringText, { children: `/${value.source}/${value.flags}` });
    }
    switch (typeof value) {
        case 'string':
            return _jsx(StringText, { children: JSON.stringify(value) });
        case 'number':
            return _jsx(NumberText, { children: JSON.stringify(value) });
        case 'boolean':
            return _jsx(BooleanText, { children: JSON.stringify(value) });
        case 'function':
            return _jsx(FunctionText, { children: "[Function]" });
        default:
            return _jsx(DefaultText, { children: JSON.stringify(value) });
    }
}
export default Inspect;
const styles = StyleSheet.create({
    spacingLeft: { marginLeft: 20 },
    expanded: { marginLeft: 20 },
    row: { paddingBottom: 8, flexDirection: 'row', alignItems: 'center' },
});
